---
title: "EWCE_eQTL_coloc"
author: "Aydan Askarova"
date: "15/03/2023"
output: html_document
---

```{r setup, include=FALSE}
#install.packages("remotes")
#remotes::install_github("NathanSkene/EWCE")

library("EWCE")
packageVersion("EWCE") #v ‘1.11.5’
library(GenomicRanges)
library(gprofiler2)
library(readxl)
library(reshape2)
library(plyr)
library(dplyr)
library(data.table)
library(pheatmap)

```

##Running EWCE 
###All the genes
```{r}
#Cell names and disease names 
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxyproxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

#Directories to save the files 
dir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/"
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/"

#ctd<-readRDS(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/ctd/ctd_Zeisel2018.rds", ) #v4
ctd <- ewceData::ctd() #v3

dat<-c()
for(c in 1:length(cell)){
  for(t in 1:length(type)){
#Assigning a unique name to each file for each cell type for each disease
load(file = paste0(dir, cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))

#Extracting gene ensgs
genes<-diseasemat_names$gene_ensg

#Getting names from gconvert2
genes_gconvert<- gconvert(query = genes, organism = "hsapiens", 
         target="ENSG",  mthreshold = Inf, filter_na = TRUE) %>% as.data.frame()

#Removing genes that dont have corresponding names 
genes_gconvert_edited<-genes_gconvert[!grepl("None", genes_gconvert$name),]

## Load merged cortex and hypothalamus dataset generated by Karolinska institute
#ctd <- ewceData::ctd()

## Gene list
hits <- genes_gconvert_edited$name

##Setting analysis parameters 

# Use 100 bootstrap lists for speed, for publishable analysis use >=10000
reps <- 100000
# Use level 1 annotations (i.e. Interneurons)
annotLevel <- 1 

#Enrichment tests
# Bootstrap significance test, no control for transcript length and GC content 
try(full_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human",
                                                hits = hits, 
                                                reps = reps,
                                                annotLevel = annotLevel))


assign(paste0(cell[c], "_", type[t], "_ewce"), full_results) 

## Tabular results
results<-full_results$results
write.table(x = results, file = paste0(outdir, cell[c], "_", type[t], "_ewce_v5.txt"), quote=F,
              row.names=F, col.names=T, sep="\t")
}
}
```

###Control EWCE

```{r required files}
#Loading genes - AD
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/AD_Jansen2019_fdr005_overlapping_genes_v3.rda")

#Loading genes - PD
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/PD_Nalls2019proxy_fdr005_overlapping_genes_v3.rda")

#Loading genes - MS
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/MS_Andlauer2016_fdr005_overlapping_genes_v3.rda")

#Loading genes - ALS
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/ALS_Rheenen2021_fdr005_overlapping_genes_v3.rda")

#Loading genes - SCZ
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/SCZ_Trubetskoy2022_fdr005_overlapping_genes_v3.rda")

#Loading genes - SCZ_Trubetskoy2022info9
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/SCZ_Trubetskoy2022info9_fdr005_overlapping_genes_v3.rda")

#Loading genes - SCZ_pardinas2018info9
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, neun_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

save(overlapping_min, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/SCZ_pardinas2018info9_fdr005_overlapping_genes_v3.rda")

```

```{r running ewce}
#Cell names and disease names 
type<-c("AD_Jansen2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022") #.v3
#Running ewce on overlapping genes as control
ctd<-readRDS(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/ctd/ctd_Zeisel2018.rds", ) #v4
#ctd <- ewceData::ctd() v3


dat<-c()
for(t in 1:length(type)){
load(file = paste0("/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/", type[t], "_fdr005_overlapping_genes_v3.rda"))
  
genes<-overlapping_min$gene_ensg

#Getting names from gconvert2
genes_gconvert<- gconvert(query = genes, organism = "hsapiens", 
         target="ENSG",  mthreshold = Inf, filter_na = TRUE) %>% as.data.frame()

#Removing genes that dont have corresponding names 
genes_gconvert_edited<-genes_gconvert[!grepl("None", genes_gconvert$name),]

## Load merged cortex and hypothalamus dataset generated by Karolinska institute
#ctd <- ewceData::ctd()

## Gene list
hits <- genes_gconvert_edited$name
##Setting analysis parameters 

# Use 100 bootstrap lists for speed, for publishable analysis use >=10000
reps <- 100000
# Use level 1 annotations (i.e. Interneurons)
annotLevel <- 1 

#Enrichment tests
# Bootstrap significance test, no control for transcript length and GC content 
try(full_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human",
                                                hits = hits, 
                                                reps = reps,
                                                annotLevel = annotLevel))
## Tabular results
results<-full_results$results
write.table(x = results, file = paste0("/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/", type[t], "_overlapping_ewce_v3.txt"), quote=F, row.names=F, col.names=T, sep="\t")
}

```

###Unique genes 
```{r - making unique gene files}
#Loading genes - AD
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_AD_Jansen2019_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_AD_Jansen2019_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_AD_Jansen2019_fdr005_unique_genes_v3.rda")

#Loading genes - PD
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_PD_Nalls2019proxy_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_PD_Nalls2019proxy_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_PD_Nalls2019proxy_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_PD_Nalls2019proxy_fdr005_unique_genes_v3.rda")

#Loading genes - MS
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_MS_Andlauer2016_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_MS_Andlauer2016_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_MS_Andlauer2016_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_MS_Andlauer2016_fdr005_unique_genes_v3.rda")

#Loading genes - ALS
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_ALS_Rheenen2021_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_ALS_Rheenen2021_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_ALS_Rheenen2021_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_ALS_Rheenen2021_fdr005_unique_genes_v3.rda")

#Loading genes - SCZ
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022_fdr005_unique_genes_v3.rda")

#Loading genes - SCZ_Trubetskoy2022info9
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022info9_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_Trubetskoy2022info9_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_Trubetskoy2022info9_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_Trubetskoy2022info9_fdr005_unique_genes_v3.rda")

#Loading genes - SCZ_pardinas2018info9
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_pardinas2018info9_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
unique_genes<-anti_join(x = pu1_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_SCZ_pardinas2018info9_fdr005_unique_genes_v3.rda")

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = olig2_only, y = neun, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_SCZ_pardinas2018info9_fdr005_unique_genes_v3.rda")

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
unique_genes<-anti_join(x = neun_only, y = olig2, "gene_ensg")
unique_genes$gene_chr<-sub("^","chr",unique_genes$gene_chr)
save(unique_genes, file="/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_SCZ_pardinas2018info9_fdr005_unique_genes_v3.rda")
```

```{r}
#Cell names and disease names 
type<-c("AD_Jansen2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022info9") #.v3
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

#Directories to save the files 
dir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/"
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/"

ctd<-readRDS(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/ctd/ctd_Zeisel2018.rds", ) #v4
#ctd <- ewceData::ctd() v3
dat<-c()
for(c in 1:length(cell)){
  for(t in 1:length(type)){
#Assigning a unique name to each file for each cell type for each disease
load(file = paste0(dir, cell[c], "_",  type[t], "_fdr005_unique_genes_v3.rda"))

#Extracting gene ensgs
genes<-unique_genes$gene_ensg

#Getting names from gconvert2
genes_gconvert<- gconvert(query = genes, organism = "hsapiens", 
         target="ENSG",  mthreshold = Inf, filter_na = TRUE) %>% as.data.frame()

#Removing genes that dont have corresponding names 
genes_gconvert_edited<-genes_gconvert[!grepl("None", genes_gconvert$name),]

## Load merged cortex and hypothalamus dataset generated by Karolinska institute
#ctd <- ewceData::ctd()

## Gene list
hits <- genes_gconvert_edited$name

##Setting analysis parameters 

# Use 100 bootstrap lists for speed, for publishable analysis use >=10000
reps <- 100000
# Use level 1 annotations (i.e. Interneurons)
annotLevel <- 1 

#Enrichment tests
# Bootstrap significance test, no control for transcript length and GC content 
try(full_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human",
                                                hits = hits, 
                                                reps = reps,
                                                annotLevel = annotLevel))

assign(paste0(cell[c], "_", type[t], "_unique_ewce"), full_results) 

## Tabular results
results<-full_results$results
write.table(x = results, file = paste0(outdir, cell[c], "_", type[t], "_unique_ewce_v3.txt"), quote=F,
              row.names=F, col.names=T, sep="\t")
}
}
```

##Merging the tables per disease
###All genes
```{r}
#Merging the tables per disease 

#AD
Microglia_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_AD_Jansen2019_ewce_v5.txt", header=T, sep="\t")
ad_pu1 = data.frame(Microglia_interactome_AD_Jansen2019_ewce,
                       list="Microglia")
Neuronal_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_AD_Jansen2019_ewce_v5.txt", header=T, sep="\t")
ad_neun = data.frame(Neuronal_interactome_AD_Jansen2019_ewce,
                        list="Neurons")
Oligo_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_AD_Jansen2019_ewce_v5.txt", header=T, sep="\t")
ad_olig = data.frame(Oligo_interactome_AD_Jansen2019_ewce,
                        list="Oligodendrocytes")

ad_merged = rbind(ad_pu1, ad_neun, ad_olig)

#AD - Kunkle
Microglia_interactome_AD_Kunkle2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_AD_Kunkle2019_ewce_v5.txt", header=T, sep="\t")
ad_pu1_k = data.frame(Microglia_interactome_AD_Kunkle2019_ewce,
                       list="Microglia")
Neuronal_interactome_AD_Kunkle2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_AD_Kunkle2019_ewce_v5.txt", header=T, sep="\t")
ad_neun_k = data.frame(Neuronal_interactome_AD_Kunkle2019_ewce,
                        list="Neurons")
Oligo_interactome_AD_Kunkle2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_AD_Kunkle2019_ewce_v5.txt", header=T, sep="\t")
ad_olig_k = data.frame(Oligo_interactome_AD_Kunkle2019_ewce,
                        list="Oligodendrocytes")

ad_merged_k = rbind(ad_pu1_k, ad_neun_k, ad_olig_k)


#PD
Microglia_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_PD_Nalls2019proxy_ewce_v5.txt", header=T, sep="\t")
pd_pu1 = data.frame(Microglia_interactome_PD_Nalls2019proxy_ewce,
                       list="Microglia")

Neuronal_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_PD_Nalls2019proxy_ewce_v5.txt", header=T, sep="\t")
pd_neun = data.frame(Neuronal_interactome_PD_Nalls2019proxy_ewce,
                        list="Neurons")

Oligo_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_PD_Nalls2019proxy_ewce_v5.txt", header=T, sep="\t")
pd_olig = data.frame(Oligo_interactome_PD_Nalls2019proxy_ewce,
                        list="Oligodendrocytes")

pd_merged = rbind(pd_pu1, pd_neun, pd_olig)

#lbd
Microglia_interactome_LBD_chia2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_LBD_chia2021_ewce_v5.txt", header=T, sep="\t")
lbd_pu1 = data.frame(Microglia_interactome_LBD_chia2021_ewce,
                    list="Microglia")

Neuronal_interactome_LBD_chia2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_LBD_chia2021_ewce_v5.txt", header=T, sep="\t")
lbd_neun = data.frame(Neuronal_interactome_LBD_chia2021_ewce,
                     list="Neurons")

Oligo_interactome_LBD_chia2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_LBD_chia2021_ewce_v5.txt", header=T, sep="\t")
lbd_olig = data.frame(Oligo_interactome_LBD_chia2021_ewce,
                     list="Oligodendrocytes")

lbd_merged = rbind(lbd_pu1, lbd_neun, lbd_olig)

#MS
Microglia_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_MS_Andlauer2016_ewce_v5.txt", header=T, sep="\t")
ms_pu1 = data.frame(Microglia_interactome_MS_Andlauer2016_ewce,
                       list="Microglia")

Neuronal_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_MS_Andlauer2016_ewce_v5.txt", header=T, sep="\t")
ms_neun = data.frame(Neuronal_interactome_MS_Andlauer2016_ewce,
                        list="Neurons")

Oligo_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_MS_Andlauer2016_ewce_v5.txt", header=T, sep="\t")
ms_olig = data.frame(Oligo_interactome_MS_Andlauer2016_ewce,
                        list="Oligodendrocytes")
ms_merged = rbind(ms_pu1, ms_neun, ms_olig)


#ALS
Microglia_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_ALS_Rheenen2021_ewce_v5.txt", header=T, sep="\t")
als_pu1 = data.frame(Microglia_interactome_ALS_Rheenen2021_ewce,
                       list="Microglia")

Neuronal_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_ALS_Rheenen2021_ewce_v5.txt", header=T, sep="\t")
als_neun = data.frame(Neuronal_interactome_ALS_Rheenen2021_ewce,
                        list="Neurons")

Oligo_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_ALS_Rheenen2021_ewce_v5.txt", header=T, sep="\t")
als_olig = data.frame(Oligo_interactome_ALS_Rheenen2021_ewce,
                        list="Oligodendrocytes")
als_merged = rbind(als_pu1, als_neun, als_olig)


#SCZ
Microglia_interactome_SCZ_Trubetskoy2022_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_SCZ_Trubetskoy2022_ewce_v5.txt", header=T, sep="\t")
scz_pu1 = data.frame(Microglia_interactome_SCZ_Trubetskoy2022_ewce,
                       list="Microglia")

Neuronal_interactome_SCZ_Trubetskoy2022_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_SCZ_Trubetskoy2022_ewce_v5.txt", header=T, sep="\t")
scz_neun = data.frame(Neuronal_interactome_SCZ_Trubetskoy2022_ewce,
                        list="Neurons")

Oligo_interactome_SCZ_Trubetskoy2022_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_SCZ_Trubetskoy2022_ewce_v5.txt", header=T, sep="\t")
scz_olig = data.frame(Oligo_interactome_SCZ_Trubetskoy2022_ewce,
                        list="Oligodendrocytes")
scz_merged = rbind(scz_pu1, scz_neun, scz_olig)

#SCZ_Trubetskoy2022info9
Microglia_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_SCZ_Trubetskoy2022info9_ewce_v3.txt", header=T, sep="\t")
scz_pu1_info9 = data.frame(Microglia_interactome_SCZ_Trubetskoy2022info9_ewce,
                       list="Microglia")

Neuronal_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_SCZ_Trubetskoy2022info9_ewce_v3.txt", header=T, sep="\t")
scz_neun_info9 = data.frame(Neuronal_interactome_SCZ_Trubetskoy2022info9_ewce,
                        list="Neurons")

Oligo_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_SCZ_Trubetskoy2022info9_ewce_v3.txt", header=T, sep="\t")
scz_olig_info9 = data.frame(Oligo_interactome_SCZ_Trubetskoy2022info9_ewce,
                        list="Oligodendrocytes")
scz_merged = rbind(scz_pu1_info9, scz_neun_info9, scz_olig_info9)

#SCZ_pardinas2018info9
Microglia_interactome_SCZ_pardinas2018info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_SCZ_pardinas2018info9_ewce_v3.txt", header=T, sep="\t")
scz_pu1_par = data.frame(Microglia_interactome_SCZ_pardinas2018info9_ewce,
                       list="Microglia")

Neuronal_interactome_SCZ_pardinas2018info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_SCZ_pardinas2018info9_ewce_v3.txt", header=T, sep="\t")
scz_neun_par = data.frame(Neuronal_interactome_SCZ_pardinas2018info9_ewce,
                        list="Neurons")

Oligo_interactome_SCZ_pardinas2018info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_SCZ_pardinas2018info9_ewce_v3.txt", header=T, sep="\t")
scz_olig_par = data.frame(Oligo_interactome_SCZ_pardinas2018info9_ewce,
                        list="Oligodendrocytes")
scz_merged_par = rbind(scz_pu1_par, scz_neun_par, scz_olig_par)

```

###Unique genes 
```{r}
#Merging the tables per disease 

#AD
Microglia_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_AD_Jansen2019_unique_ewce_v3.txt", header=T, sep="\t")
ad_pu1 = data.frame(Microglia_interactome_AD_Jansen2019_ewce,
                       list="Microglia")
Neuronal_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_AD_Jansen2019_unique_ewce_v3.txt", header=T, sep="\t")
ad_neun = data.frame(Neuronal_interactome_AD_Jansen2019_ewce,
                        list="Neurons")
Oligo_interactome_AD_Jansen2019_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_AD_Jansen2019_unique_ewce_v3.txt", header=T, sep="\t")
ad_olig = data.frame(Oligo_interactome_AD_Jansen2019_ewce,
                        list="Oligodendrocytes")

ad_merged = rbind(ad_pu1, ad_neun, ad_olig)

#PD
Microglia_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_PD_Nalls2019proxy_unique_ewce_v3.txt", header=T, sep="\t")
pd_pu1 = data.frame(Microglia_interactome_PD_Nalls2019proxy_ewce,
                       list="Microglia")

Neuronal_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_PD_Nalls2019proxy_unique_ewce_v3.txt", header=T, sep="\t")
pd_neun = data.frame(Neuronal_interactome_PD_Nalls2019proxy_ewce,
                        list="Neurons")

Oligo_interactome_PD_Nalls2019proxy_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_PD_Nalls2019proxy_unique_ewce_v3.txt", header=T, sep="\t")
pd_olig = data.frame(Oligo_interactome_PD_Nalls2019proxy_ewce,
                        list="Oligodendrocytes")

pd_merged = rbind(pd_pu1, pd_neun, pd_olig)


#ALS
Microglia_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_ALS_Rheenen2021_unique_ewce_v3.txt", header=T, sep="\t")
als_pu1 = data.frame(Microglia_interactome_ALS_Rheenen2021_ewce,
                       list="Microglia")

Neuronal_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_ALS_Rheenen2021_unique_ewce_v3.txt", header=T, sep="\t")
als_neun = data.frame(Neuronal_interactome_ALS_Rheenen2021_ewce,
                        list="Neurons")

Oligo_interactome_ALS_Rheenen2021_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_ALS_Rheenen2021_unique_ewce_v3.txt", header=T, sep="\t")
als_olig = data.frame(Oligo_interactome_ALS_Rheenen2021_ewce,
                        list="Oligodendrocytes")
als_merged = rbind(als_pu1, als_neun, als_olig)

#MS
Microglia_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_MS_Andlauer2016_unique_ewce_v3.txt", header=T, sep="\t")
ms_pu1 = data.frame(Microglia_interactome_MS_Andlauer2016_ewce,
                       list="Microglia")

Neuronal_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_MS_Andlauer2016_unique_ewce_v3.txt", header=T, sep="\t")
ms_neun = data.frame(Neuronal_interactome_MS_Andlauer2016_ewce,
                        list="Neurons")

Oligo_interactome_MS_Andlauer2016_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_MS_Andlauer2016_unique_ewce_v3.txt", header=T, sep="\t")
ms_olig = data.frame(Oligo_interactome_MS_Andlauer2016_ewce,
                        list="Oligodendrocytes")
ms_merged = rbind(ms_pu1, ms_neun, ms_olig)

#SCZ_Trubetskoy2022info9
Microglia_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Microglia_interactome_SCZ_Trubetskoy2022info9_unique_ewce_v3.txt", header=T, sep="\t")
scz_pu1_info9 = data.frame(Microglia_interactome_SCZ_Trubetskoy2022info9_ewce,
                       list="Microglia")

Neuronal_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Neuronal_interactome_SCZ_Trubetskoy2022info9_unique_ewce_v3.txt", header=T, sep="\t")
scz_neun_info9 = data.frame(Neuronal_interactome_SCZ_Trubetskoy2022info9_ewce,
                        list="Neurons")

Oligo_interactome_SCZ_Trubetskoy2022info9_ewce<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/Oligo_interactome_SCZ_Trubetskoy2022info9_unique_ewce_v3.txt", header=T, sep="\t")
scz_olig_info9 = data.frame(Oligo_interactome_SCZ_Trubetskoy2022info9_ewce,
                        list="Oligodendrocytes")
scz_merged = rbind(scz_pu1_info9, scz_neun_info9, scz_olig_info9)

```


###Control genes
```{r}
#Merging the tables per disease 

#AD
ad_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/AD_Jansen2019_overlapping_ewce_v3.txt", header=T, sep="\t")
ad_merged$type<-"AD"
#PD
pd_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/PD_Nalls2019proxy_overlapping_ewce_v3.txt", header=T, sep="\t")
pd_merged$type<-"PD"
#ALS
als_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/ALS_Rheenen2021_overlapping_ewce_v3.txt", header=T, sep="\t")
als_merged$type<-"ALS"
#MS
ms_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/MS_Andlauer2016_overlapping_ewce_v3.txt", header=T, sep="\t")
ms_merged$type<-"MS"
#SCZ
scz_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/SCZ_Trubetskoy2022_overlapping_ewce_v3.txt", header=T, sep="\t")
scz_merged$type<-"SCZ"
#SCZ_Trubetskoy2022info9
scz_merged<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/SCZ_Trubetskoy2022info9_overlapping_ewce_v3.txt", header=T, sep="\t")
scz_merged$type<-"SCZ_info9"
#SCZ_pardinas2018info9
scz_merged_par<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/ewce/SCZ_pardinas2018info9_overlapping_ewce_v3.txt", header=T, sep="\t")
scz_merged_par$type<-"SCZ_par_info9"


overlapping<-rbind(ad_merged, pd_merged, als_merged, ms_merged, scz_merged, scz_merged, scz_merged_par)
```


##Saving the plots
###All genes 
```{r Plots}
#AD
sapply(ad_merged, mode)
ad_merged$sd_from_mean_0 <- ifelse(ad_merged$sd_from_mean < 0, 0, ad_merged$sd_from_mean)
ad_merged$significant <- ifelse(ad_merged$q < 0.05, "TRUE", "FALSE")

# grouped boxplot
ad<-ggplot(ad_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in AD") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "ad_original_ewce_v3.pdf", plot = ad, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#PD
pd_merged$sd_from_mean_0 <- ifelse(pd_merged$sd_from_mean < 0, 0, pd_merged$sd_from_mean)
pd_merged$significant <- ifelse(pd_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
pd<-ggplot(pd_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in PD") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "pd_original_ewce_v3.pdf", plot = pd, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#MS

ms_merged$sd_from_mean_0 <- ifelse(ms_merged$sd_from_mean < 0, 0, ms_merged$sd_from_mean)
ms_merged$significant <- ifelse(ms_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
ms<-ggplot(ms_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in MS") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "ms_original_ewce_v3.pdf", plot = ms, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#ALS
als_merged$sd_from_mean_0 <- ifelse(als_merged$sd_from_mean < 0, 0, als_merged$sd_from_mean)
als_merged$significant <- ifelse(als_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
als<-ggplot(als_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in ALS") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))


ggplot2::ggsave(filename = "als_original_ewce_v3.pdf", plot = als, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz
scz_merged$sd_from_mean_0 <- ifelse(scz_merged$sd_from_mean < 0, 0, scz_merged$sd_from_mean)
scz_merged$significant <- ifelse(scz_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz<-ggplot(scz_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_trubetsok2019_ewce_v3.pdf", plot = scz, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz_info9
scz_merged$sd_from_mean_0 <- ifelse(scz_merged$sd_from_mean < 0, 0, scz_merged$sd_from_mean)
scz_merged$significant <- ifelse(scz_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz_info9<-ggplot(scz_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_trubetskoy2022info9_ewce_v3.pdf", plot = scz_info9, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz
scz_merged_par$sd_from_mean_0 <- ifelse(scz_merged_par$sd_from_mean < 0, 0, scz_merged_par$sd_from_mean)
scz_merged_par$significant <- ifelse(scz_merged_par$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz_par<-ggplot(scz_merged_par, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_pardinas2019info9_ewce_v3.pdf", plot = scz_par, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)
```

###Unique genes 
```{r Plots}
#AD
sapply(ad_merged, mode)
ad_merged$sd_from_mean_0 <- ifelse(ad_merged$sd_from_mean < 0, 0, ad_merged$sd_from_mean)
ad_merged$significant <- ifelse(ad_merged$q < 0.05, "TRUE", "FALSE")

# grouped boxplot
ad<-ggplot(ad_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in AD") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "ad_original_unique_ewce_v3.pdf", plot = ad, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#PD
pd_merged$sd_from_mean_0 <- ifelse(pd_merged$sd_from_mean < 0, 0, pd_merged$sd_from_mean)
pd_merged$significant <- ifelse(pd_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
pd<-ggplot(pd_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in PD") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

plot
ggplot2::ggsave(filename = "pd_original_unique_ewce_v3.pdf", plot = pd, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#MS

ms_merged$sd_from_mean_0 <- ifelse(ms_merged$sd_from_mean < 0, 0, ms_merged$sd_from_mean)
ms_merged$significant <- ifelse(ms_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
ms<-ggplot(ms_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in MS") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

plot

ggplot2::ggsave(filename = "ms_original_unique_ewce_v3.pdf", plot = ms, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#ALS
als_merged$sd_from_mean_0 <- ifelse(als_merged$sd_from_mean < 0, 0, als_merged$sd_from_mean)
als_merged$significant <- ifelse(als_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
als<-ggplot(als_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in ALS") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))


ggplot2::ggsave(filename = "als_original_unique_ewce_v3.pdf", plot = als, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz
scz_merged$sd_from_mean_0 <- ifelse(scz_merged$sd_from_mean < 0, 0, scz_merged$sd_from_mean)
scz_merged$significant <- ifelse(scz_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz<-ggplot(scz_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_trubetsok2019_unique_ewce_v3.pdf", plot = scz, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz_info9
scz_merged$sd_from_mean_0 <- ifelse(scz_merged$sd_from_mean < 0, 0, scz_merged$sd_from_mean)
scz_merged$significant <- ifelse(scz_merged$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz_info9<-ggplot(scz_merged, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_trubetskoy2022info9_unique_ewce_v3.pdf", plot = scz_info9, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)

#scz
scz_merged_par$sd_from_mean_0 <- ifelse(scz_merged_par$sd_from_mean < 0, 0, scz_merged_par$sd_from_mean)
scz_merged_par$significant <- ifelse(scz_merged_par$q < 0.05, "TRUE", "FALSE")
# grouped boxplot
scz_par<-ggplot(scz_merged_par, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(list ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=45, hjust=1), legend.position="none") +
ggtitle(label = "Expression weighted cell type enrichment analysis of H-MAGMA genes in SCZ") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = list))

ggplot2::ggsave(filename = "scz_original_pardinas2019info9_unique_ewce_v3.pdf", plot = scz_par, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 7, height = 5)
```


###Control
```{r}
#AD
sapply(overlapping, mode)
overlapping$sd_from_mean_0 <- ifelse(overlapping$sd_from_mean < 0, 0, overlapping$sd_from_mean)
overlapping$significant <- ifelse(overlapping$q < 0.05, "TRUE", "FALSE")
AD<-overlapping[grepl(c("AD"), overlapping$type),]
PD<-overlapping[grepl(c("PD"), overlapping$type),]
ALS<-overlapping[grepl(c("ALS"), overlapping$type),]
MS<-overlapping[grepl(c("MS"), overlapping$type),]
SCZ<-overlapping[grepl(c("SCZ_info9"), overlapping$type),]
overlapping<-rbind(AD, PD, MS, ALS, SCZ)
# grouped boxplot
control<-ggplot(overlapping, aes(x=CellType, y=sd_from_mean_0, fill=CellType)) + 
  geom_bar(stat="identity") +
  xlab("Cell type") +
  ylab("Std.Devs. from the mean") +
  guides(fill=guide_legend(title="Cell type")) +
  scale_fill_manual(values=c("orange", "purple", 'seagreen', "orangered3", 'deepskyblue3', 'seagreen', 'seagreen')) +
   facet_grid(type ~ .) +
  theme_bw() +
  theme(axis.text.x  =element_text(angle=30, hjust=1), legend.position="none", text = element_text(size=15)) +
ggtitle(label = "Expression weighted cell type enrichment analysis of mutual H-MAGMA genes between cell types") +
   geom_text(aes(label = ifelse(significant, "*", ""), group = type))

ggplot2::ggsave(filename = "overlapping_control_ewce_v3.pdf", plot = control, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/", width = 12, height = 8)

```

##eQTL colocalisation 
```{r - Dividing into disease specific files }
#reading the coloc file
coloc <- read_excel(path = "/Volumes/aa19618/home/HMAGMA/eQTL_Hugland/Suppl.Table.4_COLOC_Results.xlsx", sheet = "Suppl.Table.4", skip = 2, col_names = T)
#rplitting by disease type
coloc_split <- split(coloc, f = coloc$GWAS)
#dividing into disease specific files 
coloc_disease_specific<- coloc_split$AD
save(coloc_disease_specific, file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/AD_COLOC_hugland_results.rda")

coloc_disease_specific<- coloc_split$PD
save(coloc_disease_specific, file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/PD_COLOC_hugland_results.rda")

coloc_disease_specific<- coloc_split$MS
save(coloc_disease_specific, file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/MS_COLOC_hugland_results.rda")

coloc_disease_specific<- coloc_split$SCZ
save(coloc_disease_specific, file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/SCZ_COLOC_hugland_results.rda")
```


```{r - Dividing into cell type specific files}

cell<-c("Microglia", "ODC", "OPC", "Excitatory", "Inhibitory", "Astrocytes")
type<-c("AD", "PD", "MS", "SCZ")

for(c in 1:length(cell)){
  for(t in 1:length(type)){
load(file = paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/", type[t], "_COLOC_hugland_results.rda"))

coloc_cell_specific<-coloc_disease_specific[grepl(pattern = paste0(cell[c]), x = coloc_disease_specific$`Cell-type`),]

#save(coloc_disease_specific, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/", cell[c], "_", type[t], "_coloc.rda"))

assign(paste0(cell[c], "_", type[t], "_coloc"), coloc_cell_specific) 
  }
}
#AD - cells 
cell_specific_coloc<-Microglia_AD_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Microglia_interactome_AD_Jansen2019_coloc.rda"))
cell_specific_coloc<-Astrocytes_AD_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/astrocytes_AD_Jansen2019_coloc.rda"))
cell_specific_coloc<-rbind(Inhibitory_AD_coloc, Excitatory_AD_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Neuronal_interactome_AD_Jansen2019_coloc.rda"))
cell_specific_coloc<-rbind(ODC_AD_coloc, OPC_AD_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Oligo_interactome_AD_Jansen2019_coloc.rda"))

#PD - cells 
cell_specific_coloc<-Microglia_PD_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Microglia_interactome_PD_Nalls2019proxy_coloc.rda"))
cell_specific_coloc<-Astrocytes_PD_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/astrocytes_PD_Nalls2019proxy_coloc.rda"))
cell_specific_coloc<-rbind(Inhibitory_PD_coloc, Excitatory_PD_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Neuronal_interactome_PD_Nalls2019proxy_coloc.rda"))
cell_specific_coloc<-rbind(ODC_PD_coloc, OPC_PD_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Oligo_interactome_PD_Nalls2019proxy_coloc.rda"))

#MS - cells
cell_specific_coloc<-Microglia_MS_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Microglia_interactome_MS_Andlauer2016_coloc.rda"))
cell_specific_coloc<-Astrocytes_MS_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/astrocytes_MS_Andlauer2016_coloc.rda"))
cell_specific_coloc<-rbind(Inhibitory_MS_coloc, Excitatory_MS_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Neuronal_interactome_MS_Andlauer2016_coloc.rda"))
cell_specific_coloc<-rbind(ODC_MS_coloc, OPC_MS_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Oligo_interactome_MS_Andlauer2016_coloc.rda"))

#SCZ
cell_specific_coloc<-Microglia_SCZ_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Microglia_interactome_SCZ_Trubetskoy2022_coloc.rda"))
cell_specific_coloc<-Astrocytes_SCZ_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/astrocytes_SCZ_Trubetskoy2022_coloc.rda"))
cell_specific_coloc<-rbind(Inhibitory_SCZ_coloc, Excitatory_SCZ_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Neuronal_interactome_SCZ_Trubetskoy2022_coloc.rda"))
cell_specific_coloc<-rbind(ODC_SCZ_coloc, OPC_SCZ_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Oligo_interactome_SCZ_Trubetskoy2022_coloc.rda"))

#SCZ trubetskoy info9

cell_specific_coloc<-Microglia_SCZ_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Microglia_interactome_SCZ_Trubetskoy2022info9_coloc.rda"))
cell_specific_coloc<-Astrocytes_SCZ_coloc
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/astrocytes_SCZ_Trubetskoy2022info9_coloc.rda"))
cell_specific_coloc<-rbind(Inhibitory_SCZ_coloc, Excitatory_SCZ_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Neuronal_interactome_SCZ_Trubetskoy2022info9_coloc.rda"))
cell_specific_coloc<-rbind(ODC_SCZ_coloc, OPC_SCZ_coloc)
save(cell_specific_coloc, file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/Oligo_interactome_SCZ_Trubetskoy2022info9_coloc.rda"))

```

##Overlapping genes between HMAGMA and eQTL coloc
```{r}
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
type<-c("AD_Jansen2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022info9") 

dat<-c()
for(c in 1:length(cell)){
  for(t in 1:length(type)){
    
#Assigning a unique name to each file for each cell type for each disease
try(load(file = paste0("/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/", cell[c], "_",  type[t], "_fdr005_output_genes_v3.rda")))
try(load(file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/eQTL_Hugland_split/", cell[c], "_", type[t], "_coloc.rda")))  

try(overlapping_genes<-intersect(x = diseasemat_names$gene_name, y = cell_specific_coloc$Gene) %>% as.data.frame())
try(colnames(overlapping_genes)[1]<-"gene_name")
try(overlapping_genes<-join(x=overlapping_genes, y=diseasemat_names, "gene_name", type="inner", match="all"))
try(overlapping_genes<-overlapping_genes[,c(1,10)])
try(overlapping_genes$cell_type<-paste0(cell[c]))
try(overlapping_genes$disease_type<-paste0(type[t]))

try(assign(paste0(cell[c], "_", type[t], "_overlapping"), overlapping_genes))
  }
}

overlapping_genes_all<-rbind(Microglia_interactome_AD_Jansen2019_overlapping, Neuronal_interactome_AD_Jansen2019_overlapping, Oligo_interactome_AD_Jansen2019_overlapping,
                             Microglia_interactome_MS_Andlauer2016_overlapping, Neuronal_interactome_MS_Andlauer2016_overlapping, Oligo_interactome_MS_Andlauer2016_overlapping,
                               Microglia_interactome_PD_Nalls2019proxy_overlapping, Neuronal_interactome_PD_Nalls2019proxy_overlapping, Oligo_interactome_PD_Nalls2019proxy_overlapping, 
                               Microglia_interactome_SCZ_Trubetskoy2022info9_overlapping, Neuronal_interactome_SCZ_Trubetskoy2022info9_overlapping, Oligo_interactome_SCZ_Trubetskoy2022info9_overlapping)

overlapping_genes_all$log10_FDR<--log10(overlapping_genes_all$gene_fdr)
write.table(x = overlapping_genes_all, file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/eqtl/overlapping_genes_eqtl_hugland_v3.txt", col.names = T, row.names = F, sep="\t", quote = F)

```

```{r}
library(ggpubr)
library(ggbreak)

overlapping_genes_all<-read.table(file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/eqtl/overlapping_genes_eqtl_hugland_v3.txt", header = T, sep = "\t")

  scale_x_continuous(breaks = c(0,1,2,2,4,5,6,7,8,9,10,11))
ggplot(overlapping_genes_all, aes(x=disease_type, y=log10_FDR, color=cell_type)) + 
    geom_point() +
  theme_classic()+
  scale_color_manual(values=c('orangered3', 'seagreen', 'deepskyblue3'))+
  xlab("Disease type") +
  ylab("-log10(FDR corrected enrichment (q)") +
  theme(text = element_text(size=12))+
  theme(legend.position = "right",legend.direction = "vertical", legend.title = element_blank(), axis.text.x = element_text(angle = 15, vjust = 1, hjust=1), strip.background = element_blank()) 
scale_y_break(breaks = c(25,37), scales=0.2)

ggsave(filename = "eqtl_coloc_hmagma_plot_v2.pdf", plot = last_plot(), device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/",  width = 16, height = 15, units = "cm")


```

##enhancers from unique gene regions for ldsc
```{r UNIQUE GENES}
load ("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/exon_promoranges.rda") #exonranges, promoterranges

load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Microglia_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
pu1<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Oligo_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
olig2<-diseasemat_names
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/Neuronal_interactome_AD_Jansen2019_fdr005_output_genes_v3.rda")
neun<-diseasemat_names

pu1_only<-anti_join(x = pu1, y = olig2, "gene_ensg")
pu1_only<-anti_join(x = pu1_only, y = neun, "gene_ensg")
pu1_only$gene_chr<-sub("^","chr",pu1_only$gene_chr)
pu1_only_grages<-GRanges(seqnames = pu1_only$gene_chr, ranges = IRanges(start = pu1_only$gene_start, end = pu1_only$gene_end), gene_name=pu1_only$gene_name, gene_ensg=pu1_only$gene_ensg)

olig2_only<-anti_join(x = olig2, y = pu1, "gene_ensg")
olig2_only<-anti_join(x = olig2_only, y = neun, "gene_ensg")
olig2_only_grages<-GRanges(seqnames = olig2_only$gene_chr, ranges = IRanges(start = olig2_only$gene_start, end = olig2_only$gene_end), strand = olig2_only$gene_strand, gene_name=olig2_only$gene_name, gene_ensg=olig2_only$gene_ensg)

neun_only<-anti_join(x = neun, y = pu1, "gene_ensg")
neun_only<-anti_join(x = neun_only, y = olig2, "gene_ensg")
neun_only_grages<-GRanges(seqnames = neun_only$gene_chr, ranges = IRanges(start = neun_only$gene_start, end = neun_only$gene_end), strand = neun_only$gene_strand, gene_name=neun_only$gene_name, gene_ensg=neun_only$gene_ensg)

all_genes<-rbind(pu1, olig2, neun)
all_unique_genees<-rbind(pu1_only, olig2_only, olig2_only)
overlapping_min<-anti_join(x = all_genes, y = all_unique_genees, "gene_ensg")
overlapping_min<-join(x = overlapping_min, y = pu1, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = olig2, "gene_ensg", type="inner")
overlapping_min<-join(x = overlapping_min, y = neun, "gene_ensg", type="inner")

```

```{r}
#Reading the plac-seq files for each cell type
hic<- read_excel(path = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/PLAC/aay0793-nott-table-s5.xlsx", sheet = "Microglia_interactome", skip = 2, col_names = T)
    
    #Doubling the hic and flipping the start1-end1 with int1-int2 (this is for GRanges function and IRanges)
    hic.int1 <- hic[,1:6]
    hic.int2 <- hic[,c(4:6,1:3)]
    colnames(hic.int1) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    colnames(hic.int2) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    hic <- rbind(hic.int1, hic.int2)
    
    #Creating GRanges for findOverlaps() function
    hicranges <- GRanges(hic$chrom1, IRanges(as.numeric(hic$start1), as.numeric(hic$end1)),
                        int1=hic$start2,int2=hic$end2)
    
    #Reading promoters data (Nott et al., 2019 paper)
    promoters<-read.table(file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/Microglia_interactome_promoters.bed"), header=F, sep="\t")
    colnames(promoters)<-c("chr", "start", "end")
    promoters_ranges<-GRanges(promoters$chr, IRanges(as.numeric(promoters$start), as.numeric(promoters$end)))
  
    #Selecting plac-seq interactions that overlap with promoters from Chip-seq (this is just in case the interactions pulled by plac-seq are not tied to promoters)
    olap<-findOverlaps(query = hicranges, subject = promoters_ranges) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    placranges<-hicranges[queryHits(olap)]
    mcols(placranges)<-cbind(mcols(hicranges[queryHits(olap)]), mcols(promoters_ranges[subjectHits(olap)])) 

#selecting g ranges that contain the risk genes
    olap<-findOverlaps(query = placranges, subject = pu1_only_grages) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    placranges_pu1_pu1_genes<-placranges[queryHits(olap)]
    mcols(placranges_pu1_pu1_genes)<-cbind(mcols(hicranges[queryHits(olap)]), mcols(promoters_ranges[subjectHits(olap)])) 
  placranges_pu1_pu1_genes<-as.data.frame(placranges_pu1_pu1_genes)
  placranges_pu1_pu1_genes<-unique(placranges_pu1_pu1_genes)
    placranges_pu1_pu1_genes_interacting<-GRanges(seqnames = placranges_pu1_pu1_genes$seqnames, ranges = IRanges(start = placranges_pu1_pu1_genes$int1, end = placranges_pu1_pu1_genes$int2), gene_start = placranges_pu1_pu1_genes$start, gene_end=placranges_pu1_pu1_genes$end)
    
    #Reading enhancers data  (Nott et al., 2019 paper) 
    enhancers<-read.table(file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/Microglia_interactome_enhancers.bed"), header=F, sep="\t")
    colnames(enhancers)<-c("chr", "start", "end")
   enhancers_ranges<-GRanges(enhancers$chr, IRanges(as.numeric(enhancers$start), as.numeric(enhancers$end)))   
  
    #Selecting exonic/promoter interacting SNPS overlapping with enhancers   
    olap<-findOverlaps(query = enhancers_ranges, subject = placranges_pu1_pu1_genes_interacting) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    placranges_pu1_pu1_genes_interacting_ranges<-enhancers_ranges[queryHits(olap)]
    mcols(placranges_pu1_pu1_genes_interacting_ranges)<-cbind(mcols(enhancers_ranges[queryHits(olap)]), mcols(placranges_pu1_pu1_genes_interacting[subjectHits(olap)]))

    #Making a dataframe with exonic/promoter interacting enhancer SNPs and ONLY THEN removing row duplicates 
    placranges_pu1_pu1_genes_interacting_ranges<-as.data.frame(placranges_pu1_pu1_genes_interacting_ranges) #•
    placranges_pu1_pu1_genes_interacting_ranges<-unique(placranges_pu1_pu1_genes_interacting_ranges) #• 
```


##Enrichment point graph
###Control
```{r}
install.packages("ggbreak")
library(ggbreak)

type<-c("AD_Jansen2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022info9") #.v3
dat<-c()
for(t in 1:length(type)){
load(file = paste0("/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/", type[t], "_fdr005_overlapping_genes_v3.rda"))
assign(paste0(type[t], "_control_genes"), overlapping_min)
}
control<-rbind(AD_Jansen2019_control_genes, PD_Nalls2019proxy_control_genes, MS_Andlauer2016_control_genes,ALS_Rheenen2021_control_genes )
control<-control[grepl(c("Microglia"), control$cell),]
control<-control[,c(2,10,13)]
control$log10_FDR_pu1<--log10(control$gene_fdr)
control<-unique(control)

# grouped boxplot
control_plot<-ggplot(control, aes(x=disease, y=log10_FDR_pu1, fill=disease)) + 
    geom_point() +
  theme_classic()+
  scale_color_manual(values=c('orangered3', 'seagreen', 'deepskyblue3', 'yellow', 'orange'))+
  geom_text(aes(label=gene_name),hjust=1, vjust=1, size=10, angle=0)+
  scale_y_continuous(limits=c(0,12))+
  xlab("Disease type") +
  ylab("-log10(FDR corrected enrichment (q)") +
  theme(text = element_text(size=30))+
  theme(legend.position = "none", axis.text.x = element_text(angle = 15, vjust = 1, hjust=1), strip.background = element_blank())
control_plot
ggplot2::ggsave(filename = "control_genes_points_v3.pdf", plot = control_plot, device = "pdf", path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/genes/", width = 15, height = 25)

```
###Unique genes
```{r}
#Directories to save the files 
dir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/"
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/"

type<-c("AD_Jansen2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "SCZ_Trubetskoy2022info9", "SCZ_pardinas2018info9") #.v3
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

for(t in 1:length(type)){
  for(c in 1)
load(file = paste0(dir, cell[c], "_",  type[t], "_fdr005_unique_genes_v3.rda"))
}
```

```{r}
ad_merged<-ad_merged[order(ad_merged$CellType), ]
ad_merged$log10_q= -log10(ad_merged$q)
ad_merged$log10_q<-gsub("Inf", 10, ad_merged$log10_q)
ad_merged_split<-ad_merged[,c(1,5,7,8)]
ad_merged_split<-split(x =ad_merged_split, f = ad_merged_split$list )
microglia<-transpose(ad_merged_split$Microglia)
neurons<-transpose(ad_merged_split$Neurons)
olig<-transpose(ad_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
AD<-rbind(microglia,neurons,olig)
names(AD)<-ad_merged_split$Microglia$CellType
rownames(AD)<-c("Microglia_AD", "Neurons_AD", "Oligodendrocytes_AD")

ad_merged_split<-transpose(ad_merged_split)
ad_merged_split<-ad_merged_split[2,]
rownames(ad_merged_split)<-"AD"

ad_merged_k<-ad_merged_k[order(ad_merged_k$CellType), ]
ad_merged_k$log10_q= -log10(ad_merged_k$q)
ad_merged_k$log10_q<-gsub("Inf", 10, ad_merged_k$log10_q)
ad_merged_split_k<-ad_merged_k[,c(1,5,7,8)]
ad_merged_split_k<-split(x =ad_merged_split_k, f = ad_merged_split_k$list )
microglia<-transpose(ad_merged_split_k$Microglia)
neurons<-transpose(ad_merged_split_k$Neurons)
olig<-transpose(ad_merged_split_k$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
AD_k<-rbind(microglia,neurons,olig)
names(AD_k)<-ad_merged_split_k$Microglia$CellType
rownames(AD_k)<-c("Microglia_AD_k", "Neurons_AD_k", "Oligodendrocytes_AD_k")

pd_merged<-pd_merged[order(pd_merged$CellType), ]
pd_merged$log10_q= -log10(pd_merged$q)
pd_merged_split<-pd_merged[,c(1,5,7,8)]
pd_merged_split<-split(x =pd_merged_split, f = pd_merged_split$list )
microglia<-transpose(pd_merged_split$Microglia)
neurons<-transpose(pd_merged_split$Neurons)
olig<-transpose(pd_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
PD<-rbind(microglia,neurons,olig)
names(PD)<-pd_merged_split$Microglia$CellType
rownames(PD)<-c("Microglia_PD", "Neurons_PD", "Oligodendrocytes_PD")

lbd_merged<-lbd_merged[order(lbd_merged$CellType), ]
lbd_merged$log10_q= -log10(lbd_merged$q)
lbd_merged_split<-lbd_merged[,c(1,5,7,8)]
lbd_merged_split<-split(x =lbd_merged_split, f = lbd_merged_split$list )
microglia<-transpose(lbd_merged_split$Microglia)
neurons<-transpose(lbd_merged_split$Neurons)
olig<-transpose(lbd_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
LBD<-rbind(microglia,neurons,olig)
names(LBD)<-lbd_merged_split$Microglia$CellType
rownames(LBD)<-c("Microglia_LBD", "Neurons_LBD", "Oligodendrocytes_LBD")


pd_merged_split<-transpose(pd_merged_split)
pd_merged_split<-pd_merged_split[2,]
rownames(pd_merged_split)<-"PD"

ms_merged<-ms_merged[order(ms_merged$CellType), ]
ms_merged$log10_q= -log10(ms_merged$q)
ms_merged$log10_q<-gsub("Inf", 10, ms_merged$log10_q)
ms_merged_split<-ms_merged[,c(1,5,7,8)]
ms_merged_split<-split(x =ms_merged_split, f = ms_merged_split$list )
microglia<-transpose(ms_merged_split$Microglia)
neurons<-transpose(ms_merged_split$Neurons)
olig<-transpose(ms_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
MS<-rbind(microglia,neurons,olig)
names(MS)<-ms_merged_split$Microglia$CellType
rownames(MS)<-c("Microglia_MS", "Neurons_MS", "Oligodendrocytes_MS")

ms_merged_split<-transpose(ms_merged_split)
ms_merged_split<-ms_merged_split[2,]
rownames(ms_merged_split)<-"MS"

als_merged<-als_merged[order(als_merged$CellType), ]
als_merged$log10_q= -log10(als_merged$q)
als_merged_split<-als_merged[,c(1,5,7,8)]
als_merged_split<-split(x =als_merged_split, f = als_merged_split$list )
microglia<-transpose(als_merged_split$Microglia)
neurons<-transpose(als_merged_split$Neurons)
olig<-transpose(als_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
ALS<-rbind(microglia,neurons,olig)
names(ALS)<-als_merged_split$Microglia$CellType
rownames(ALS)<-c("Microglia_ALS", "Neurons_ALS", "Oligodendrocytes_ALS")

als_merged_split<-transpose(als_merged_split)
als_merged_split<-als_merged_split[2,]
rownames(als_merged_split)<-"ALS"

scz_merged<-scz_merged[order(scz_merged$CellType), ]
scz_merged$log10_q= -log10(scz_merged$q)
scz_merged$log10_q<-gsub("Inf", 10, scz_merged$log10_q)
scz_merged_split<-scz_merged[,c(1,5,7,8)]
scz_merged_split<-split(x =scz_merged_split, f = scz_merged_split$list )
microglia<-transpose(scz_merged_split$Microglia)
neurons<-transpose(scz_merged_split$Neurons)
olig<-transpose(scz_merged_split$Oligodendrocytes)
microglia<-microglia[4,]
neurons<-neurons[4,]
olig<-olig[4,]
SCZ<-rbind(microglia,neurons,olig)
names(SCZ)<-scz_merged_split$Microglia$CellType
rownames(SCZ)<-c("Microglia_SCZ", "Neurons_SCZ", "Oligodendrocytes_SCZ")

scz_merged_split<-transpose(scz_merged_split)
scz_merged_split<-scz_merged_split[2,]
rownames(scz_merged_split)<-"SCZ"

all_disease<-rbind(AD, AD_k, PD, MS, ALS, SCZ)

all_disease<-rbind(ad_merged_split, pd_merged_split, ms_merged_split, als_merged_split, scz_merged_split)
colnames(all_disease)<-ad_merged$CellType

sapply(all_disease, mode)

all_disease<-transform(all_disease, Glia=as.numeric(Glia),
                      Immune_cells=as.numeric(Immune_cells),
                      Neurons=as.numeric(Neurons),
                      Vascular_cells=as.numeric(Vascular_cells))
all_disease<-all_disease[,c(2,3,1,4)]

all_disease<-transform(all_disease, microglia=as.numeric(microglia),
                       endothelial_mural=as.numeric(endothelial_mural),
                       oligodendrocytes=as.numeric(oligodendrocytes),
                       interneurons=as.numeric(interneurons),
                       pyramidal_SS=as.numeric(pyramidal_SS),
                       pyramidal_CA1=as.numeric(pyramidal_CA1),
                       astrocytes_ependymal=as.numeric(astrocytes_ependymal))
all_disease<-all_disease[,c(4,3,6,7,5,1,2)]
all_disease_transposed<-transpose(all_disease)
colnames(all_disease_transposed)<-rownames(all_disease)
rownames(all_disease_transposed)<-colnames(all_disease)

```

##For UNIQUE/ALL GENES PHEATMAP
```{r}
annotation_col<-data.frame(CellType=c(rep(c("Microglia", "Neurons","Oligodendrocytes"), times=6)),
                           DiseaseType= c(rep(c("AD"), times=3), rep(c("AD_Kunkle"), times=3), rep(c("PD"), times=3), rep(c("MS"), times=3),rep(c("ALS"), times=3),rep(c("SCZ"), times=3)))

rownames(annotation_col)<-colnames(all_disease_transposed)

ann_colors = list(CellType = c(Microglia="orangered3", Neurons="seagreen", Oligodendrocytes="deepskyblue3"), DiseaseType = c(AD="darkorange2", AD_Kunkle = "goldenrod1",  PD="slateblue3", MS="midnightblue", ALS="orangered4", SCZ="olivedrab3"))

all_disease_transposed<-as.matrix(all_disease_transposed)
ewce_heatmap<-pheatmap(mat = all_disease_transposed, 
        color = c("white","pink1", "palevioletred1", "hotpink3", "darkorchid4"), 
        breaks = c(1, 1.30103, 2, 3, 4, 5),
        display_numbers = T, 
        number_color = c("black"), 
        fontsize_number = 10, 
        number_format =  "%.1f",
        cluster_rows = F, 
        cluster_cols = F, 
        gaps_row = c(1,2,3,4,5,6),
        gaps_col = c(3,6,9,12,15),
        border_color = "grey88",
        annotation_col = annotation_col,
        annotation_colors = ann_colors, 
        annotation_names_row = T, 
        show_rownames = T,
        show_colnames = F, 
        cellwidth = 35, 
        cellheight = 30,
        fontsize_row = 10, 
        fontsize_col = 10, 
        height = 10, 
        width = 14,
        main = c("EWCE of all HMAGMA genes to each cell type")

-log10(p.adjust(p = 0.05, method = "BH")) #1.30103
?coverage
```
##For CONTROL GENES PHEATMAP
```{r}
annotation_row<-data.frame(list=c("AD", "PD", "MS", "ALS", "SCZ"))

rownames(annotation_row)<-rownames(annotation_row)

ann_colors = list(list = c(AD="orangered3", PD="seagreen", MS="deepskyblue3", ALS="pink", SCZ="purple"))



breaksList = seq(0, 5, by = 1)

pheatmap(all_disease, 
        color = colorRampPalette(brewer.pal(n = 7, name = "Blues"))(length(breaksList)), breaks = breaksList,
        display_numbers = T, 
        border_color = "white",
        gaps_row = c(1,2,2,4,5),
        gaps_col = c(1,2,2,4,5,6,7),
        annotation_row = annotation_row,
        annotation_colors = ann_colors, 
        annotation_names_row = T, 
        show_rownames = T,
        show_colnames = T, 
        angle_col = 45,
        cluster_rows = F, 
        cluster_cols = F, 
        number_color = c("black"), 
        fontsize_number = 5, 
        number_format =  "%.2f",
        cellwidth = 12, 
        cellheight = 10,
        fontsize = 7, 
        fontsize_row = 7, 
        fontsize_col = 7, 
        height = 5, 
        width = 7.5,
        main = c("Expression Weighted Cell type Enrichment of HMAGMA overlapping genes to each cell type"),
        filename = "/Volumes/aa19618/home/HMAGMA_Protocol/results/graphs/ewce/pheatmap_overlapping_genes_significance_log10.pdf")

```






