
```{r}
library("EWCE")
packageVersion("EWCE") #v ‘1.11.5’
library(GenomicRanges)
library(gprofiler2)
library(readxl)
library(reshape2)
library(plyr)
library(dplyr)
library(data.table)
library(pheatmap)
```

##Running EWCE 
###All the genes
```{r}
#Cell names and disease names 

load("ctd/sc_tsartsalis2024_sc_tsartsalis.rda")
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")

cell<- c("Oligo_interactome")

#Directories to save the files 
dir<-"~/HMAGMA_Protocol/output_annotation_v4/"
outdir<-"~/HMAGMA_Protocol/results/tables/"

#ctd<-readRDS(file = "~/HMAGMA_Protocol/required_files/ctd/ctd_Zeisel2018.rds", ) #v4
dat<-c()
for(c in 1:length(cell)){
  for(t in 1:length(type)){
#Assigning a unique name to each file for each cell type for each disease
load(file = paste0("~/HMAGMA_Protocol/output_annotation_v4/", cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))

#Extracting gene ensgs
genes<-diseasemat_names$gene_ensg

#Getting names from gconvert2
genes_gconvert<- gconvert(query = genes, organism = "hsapiens", 
         target="ENSG",  mthreshold = Inf, filter_na = TRUE) %>% as.data.frame()

#Removing genes that dont have corresponding names 
genes_gconvert_edited<-genes_gconvert[!grepl("None", genes_gconvert$name),]

## Load merged cortex and hypothalamus dataset generated by Karolinska institute
#ctd <- ewceData::ctd()

## Gene list
hits <- genes_gconvert_edited$name

##Setting analysis parameters 

# Use 100 bootstrap lists for speed, for publishable analysis use >=10000
reps <- 100000
# Use level 1 annotations (i.e. Interneurons)
annotLevel<-c(1)

#Enrichment tests
# Bootstrap significance test, no control for transcript length and GC content 
try(full_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                sctSpecies = "human",
                                                genelistSpecies = "human",
                                                hits = hits, 
                                                reps = reps,
                                                annotLevel = annotLevel))


assign(paste0(cell[c], "_", type[t], "_ewce"), full_results) 

## Tabular results
results<-full_results$results
write.table(x = results, file = paste0(outdir, cell[c], "_", type[t], "_ewce_tsartsalis.txt"), quote=F,
              row.names=F, col.names=T, sep="\t")
}
}
```

```{r}
##Uploading the ewce data 


type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

fdr<-c(0.01)
annotType<-c("cluster_celltype_paper" = 1)
annotLevel<-c(1)

std<-data.frame(CellType=ewce_full$CellType)
pvalue<-data.frame(CellType=ewce_full$CellType)

for(c in 1:length(cell)){
  for(t in 1:length(type)){
ewce_full<-fread(input = paste0("~/HMAGMA_Protocol/results/tables/", cell[c], "_", type[t], "_ewce_tsartsalis.txt"))

results<-ewce_full[,c(1,5)]
significant<-ewce_full[,c(1,6)]
significant$q <- ifelse(
  significant$q == 0, "***",
  ifelse(
    significant$q > 0 & significant$q <= 5e-8, "**",
    ifelse(
      significant$q > 5e-8 & significant$q <= 0.05, "*", ""
    )
  )
)


colnames(results)<-c("CellType", paste0(type[t], "_", cell[c]))
colnames(significant)<-c("CellType", paste0(type[t], "_", cell[c]))

std<-merge(std, results, by="CellType")
pvalue<-merge(pvalue, significant, by="CellType")
  }}
rownames(std)<-std$CellType
std<-std[,-1]
rownames(pvalue)<-pvalue$CellType
pvalue<-pvalue[,-1]

```

##Ewce visualising
```{r}

color_palette <- colorRampPalette(colors = c("#FFFFFF", "#EAFDB4", "#C5E862", "#6DC335", "#117733"))(100)
#color_palette <- colorRampPalette(colors = c("#FFFFFF","#C5E862","#6DC335", "#117733"))(1000)

rownames(std)<-c("ASTROCYTES", "BEC", "FIBROBLASTS", "LYMPHOCYTES", "MICROGLIA", "NEURONS", "OLIGODENDROCYTES", "SMC/PERICYTES")
std<-std[c("MICROGLIA", "LYMPHOCYTES", "BEC", "FIBROBLASTS", "SMC/PERICYTES", "ASTROCYTES", "NEURONS", "OLIGODENDROCYTES"),]

colnames(std)<-gsub(pattern = "Microglia_interactome", replacement = "microglia", colnames(std))
colnames(std)<-gsub(pattern = "Oligo_interactome", replacement = "oligo", colnames(std))
colnames(std)<-gsub(pattern = "Neuronal_interactome", replacement = "neurons", colnames(std))

colnames(pvalue)<-gsub(pattern = "Microglia_interactome", replacement = "microglia", colnames(pvalue))
colnames(pvalue)<-gsub(pattern = "Oligo_interactome", replacement = "oligo", colnames(pvalue))
colnames(pvalue)<-gsub(pattern = "Neuronal_interactome", replacement = "neurons", colnames(pvalue))

std<-std[,c("AD_Jansen2019_microglia", "AD_Jansen2019_neurons", "AD_Jansen2019_oligo", 
"AD_Kunkle2019_microglia", "AD_Kunkle2019_neurons", "AD_Kunkle2019_oligo", 
"PD_Nalls2019proxy_microglia", "PD_Nalls2019proxy_neurons","PD_Nalls2019proxy_oligo", 
"MS_Andlauer2016_microglia", "MS_Andlauer2016_neurons","MS_Andlauer2016_oligo", 
"ALS_Rheenen2021_microglia",  "ALS_Rheenen2021_neurons","ALS_Rheenen2021_oligo",
"SCZ_Trubetskoy2022_microglia", "SCZ_Trubetskoy2022_neurons", "SCZ_Trubetskoy2022_oligo")]

pvalue<-pvalue[,c("AD_Jansen2019_microglia", "AD_Jansen2019_neurons", "AD_Jansen2019_oligo", 
"AD_Kunkle2019_microglia", "AD_Kunkle2019_neurons", "AD_Kunkle2019_oligo", 
"PD_Nalls2019proxy_microglia", "PD_Nalls2019proxy_neurons","PD_Nalls2019proxy_oligo", 
"MS_Andlauer2016_microglia", "MS_Andlauer2016_neurons","MS_Andlauer2016_oligo", 
"ALS_Rheenen2021_microglia",  "ALS_Rheenen2021_neurons","ALS_Rheenen2021_oligo",
"SCZ_Trubetskoy2022_microglia", "SCZ_Trubetskoy2022_neurons", "SCZ_Trubetskoy2022_oligo")]
rownames(pvalue)<-c("ASTROCYTES", "BEC", "FIBROBLASTS", "LYMPHOCYTES", "MICROGLIA", "NEURONS", "OLIGODENDROCYTES", "SMC/PERICYTES")
pvalue<-pvalue[c("MICROGLIA", "LYMPHOCYTES", "BEC", "FIBROBLASTS", "SMC/PERICYTES", "ASTROCYTES", "NEURONS", "OLIGODENDROCYTES"),]

annotation_col<-data.frame(CellType=colnames(std))
rownames(annotation_col)<-annotation_col$CellType

annotation_color<-list(CellType=c("MICROGLIA" = "#BC3934", "OLIGODENDROCYTES"="#0072B2", "NEURONS"="#009E73"))

annotation_color<-list(CellType= setNames(
  ifelse(grepl("microglia", colnames(std), ignore.case = TRUE), "#BC3934",
  ifelse(grepl("neuron", colnames(std), ignore.case = TRUE), "#009E73",
  ifelse(grepl("oligo", colnames(std), ignore.case = TRUE), "#0072B2", "black"))),  # Default color
  colnames(std)
))

std[std < 0] <- 0

pheatmap(mat = std, annotation_col = annotation_col, annotation_colors = annotation_color, show_colnames = T,
         gaps_col = c(3, 6, 9, 12, 15, 18),
         color = color_palette, 
         display_numbers = pvalue, 
         cluster_rows = F, 
         cluster_cols = F, 
         treeheight_row = F, 
         treeheight_col = F, 
         angle_col = 90, 
         fontsize_number = 12,
         fontsize = 7, cellwidth = 30, cellheight = 30, height = 5, width = 15,
         filename = paste0("graphs/ewce/ewce_tsartsalis_hmagma.pdf"))
```



