---
title: "Enhancer Interactions Sampled Down"
author: "Aydan Askarova"
date: '2022-10-04'
output: html_document
---

##Loading packages
```{r packages}
options(stingAsFactors=F)
library(GenomicRanges)
library(biomaRt)
library(dplyr)
library(readxl)
library("reshape")
library(ggplot2)
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(stringr)
```

##Loading required files 
```{r}
load ("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/exon_promoranges.rda")
load ("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda")
load("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/snp_locating_in_exon_promoter_transcript_level_g1000.rda")
load("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/non_exonic_promoter_snp_g1000.rda")
load(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/gene_information.rda")
```

##Loading PLAC-seq and sampling down 
```{r loading PLAC-seq sampled down}

#Cell names and random number generator configurations 1:10 
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
seed<- c(1:10)

#Directories to save the files 
outdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/annotation_files/"
inputdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/H3K27AC_peaks/"

#Looping
dat <- c()
for(i in 1:length(cell)) {
  for(s in 1:10) {
    
    #Reading the plac-seq file
    hic <- read_excel(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/PLAC/aay0793-nott-table-s5.xlsx", sheet = paste0(cell[i]), skip = 2, col_names = T)
    
    #Sampling 60000 random rows from the plac-seq data table using set.seed() function in data.table package 
    #(https://stackoverflow.com/questions/8273313/sample-random-rows-in-dataframe)
    set.seed(s)
    hic <- hic[sample(nrow(hic), size=60000),] 
  
    #Including all the bins by doubling and flipping the data
    hic.int1 <- hic[,1:6]
    hic.int2 <- hic[,c(4:6,1:3)]
    colnames(hic.int1) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    colnames(hic.int2) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    hic <- rbind(hic.int1, hic.int2)
    
    #Creating GRanges for findOverlaps() function
    hicranges <- GRanges(hic$chrom1, IRanges(as.numeric(hic$start1), as.numeric(hic$end1)),
                        int1=hic$start2,int2=hic$end2)
    
    #Reading promoters data  (Nott et al., 2019 paper)
    promoters<-read.table(file=paste0("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/", cell[i], "_promoters.bed"), header=F, sep="\t")
    colnames(promoters)<-c("chr", "start", "end")
    promoters_ranges<-GRanges(promoters$chr, IRanges(as.numeric(promoters$start), as.numeric(promoters$end)))
    
    #Selecting interactions that overlap with promoters 
    olap<-findOverlaps(query = hicranges, subject = promoters_ranges) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    placranges<-hicranges[queryHits(olap)]
    mcols(placranges)<-cbind(mcols(hicranges[queryHits(olap)]), mcols(promoters_ranges[subjectHits(olap)])) 

    #Selecting plac interactions that overlap with exons 
    olap <- findOverlaps(query = placranges,subject = exonranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    exonint <- placranges[queryHits(olap)];
    mcols(exonint) <- cbind(mcols(placranges[queryHits(olap)]), mcols(exonranges[subjectHits(olap)]))
    
    #Selecting plac interactions that overlap with promoters  
    olap <- findOverlaps(query = placranges,subject = promoterranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    proint <- placranges[queryHits(olap)];
    mcols(proint) <- cbind(mcols(placranges[queryHits(olap)]), mcols(promoterranges[subjectHits(olap)]))
    
    #Combining plac interactions containing exons and promoters 
    generanges <- c(exonint,proint)
    
    #Making a dataframe with plac-ranges interacting regions 
    genebed <- data.frame(chr=seqnames(generanges), snp.start=generanges$int1, snp.end=generanges$int2,
                          gene.start=start(generanges), gene.end=start(generanges)+width(generanges)-1,
                          ensg=generanges$gene)
    
    #Since some plac interactions will overlap between exons and promoters, keeping only the unique rows (removing duplicates)
    genebed <- unique(genebed) 
    genesnpranges <- GRanges(genebed$chr, IRanges(genebed$snp.start, genebed$snp.end), ensg=genebed$ensg)

    #Finding SNPs that fall within plac-ranges interacting regions
    olap <- findOverlaps(query = snpranges, subject = genesnpranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    snpint <- snpranges[queryHits(olap)];
    mcols(snpint) <- cbind(mcols(snpranges[queryHits(olap)]), mcols(genesnpranges[subjectHits(olap)]))
    
    #Since some SNPs will overlap keeping only the unique rows for all interactions (removing duplicates)
    snpint_data<-as.data.frame(snpint) #•
    snpint_data <- unique(snpint_data) #•
    
    snpint<-GRanges(seqnames = snpint_data$seqnames, ranges = IRanges(snpint_data$start, snpint_data$end), 
                    strand = snpint_data$strand, rsid=snpint_data$rsid, ensg=snpint_data$ensg)
    
    save(snpint, file=paste0(outdir, "Hi-C_transcript_interacting_snp_",cell[i],"_SAMPLED_DOWN", s,".rda"))
    
    #Reading enhancers data  (Nott et al., 2019 paper) 
    enhancers<-read.table(file=paste0("/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/", cell[i], "_enhancers.bed"), header=F, sep="\t")
    colnames(enhancers)<-c("chr", "start", "end")
    enhancers_ranges<-GRanges(enhancers$chr, IRanges(as.numeric(enhancers$start), as.numeric(enhancers$end)))   
    
    #Selecting SNPs overlaping with plac-ranges interacting enhancers   
    olap<-findOverlaps(query = snpint, subject = enhancers_ranges) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    enhancer_snp_ranges<-snpint[queryHits(olap)]
    mcols(enhancer_snp_ranges)<-cbind(mcols(snpint[queryHits(olap)]), mcols(enhancers_ranges[subjectHits(olap)]))
    
    #Removing duplicate rows
    enhancer_snp_ranges_data<-as.data.frame(enhancer_snp_ranges) #•
    enhancer_snp_ranges_data<-unique(enhancer_snp_ranges_data) #• 
  
    enhancer_snp_ranges<-GRanges(seqnames = enhancer_snp_ranges_data$seqnames, ranges = IRanges(enhancer_snp_ranges_data$start, enhancer_snp_ranges_data$end), 
                                 strand = enhancer_snp_ranges_data$strand, rsid=enhancer_snp_ranges_data$rsid, ensg=enhancer_snp_ranges_data$ensg)
    
  
    save(enhancer_snp_ranges, file=paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_",cell[i],"_SAMPLED_DOWN", s,".rda"))
}
}

```

```{r - Annotation file - exo, pro, intr, inter}
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
outdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/annotation_files/"

dat<-c()
for(i in 1:length(cell)){
  for(s in 1:10){
  load(file=paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_",cell[i],"_SAMPLED_DOWN", s,".rda"))
  snpdat <- data.frame(chr=seqnames(enhancer_snp_ranges), bp=start(enhancer_snp_ranges), rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg)
  
  snpromat <- unique(data.frame(rsid=snpro$rsid, ensg=snpro$gene))
  snpexonmat <- unique(data.frame(rsid=snpexon$rsid, ensg=snpexon$gene))
  snpgene<-unique(rbind(snpromat,snpexonmat))
  
  snpintmat<-unique(data.frame(rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg))
  
  snpcomb_loop<-merge(x = snpgene, y = snpintmat, by.x="ensg", by.y="ensg", all.y=TRUE)
  snpcomb_loop_melt<-melt(snpcomb_loop, id=c("ensg"))
  snpcomb_loop_melt<-snpcomb_loop_melt[,c(3,1)]
  snpcomb_loop_melt<-unique(snpcomb_loop_melt)
  snpcomb_loop_melt<-na.omit(snpcomb_loop_melt)
  colnames(snpcomb_loop_melt)<-c("rsid", "ensg")
  
  save(snpcomb_loop_melt, file=paste0(outdir,"SNP_to_transcript_comb_", cell[i], "_SAMPLED_DOWN", s, ".rda")) 
  
  snpagg <- aggregate(snpcomb_loop_melt, list(snpcomb_loop_melt$ensg), unique)
  load(file="/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/gencode41_37_pro_exo_gene.rda")
  colnames(gene) <- c("chr", "start", "end", "ensg")
  gene<-transform(gene, chr=as.character(chr))
  gene <- gene[grep("chr", gene$chr),]
  gene$chr <- unlist(lapply(strsplit(gene$chr, "chr"), '[[', 2))
  gene$index <- paste(gene$chr, gene$start, gene$end, sep=":")
  snpagg$index = gene[match(snpagg$ensg, gene$ensg),"index"]
  
  snpagg <- snpagg[!is.na(snpagg$index),]
  snpaggconv <- snpagg[,c("ensg", "index", "rsid")]
  writable <- format(snpaggconv)
  
  write.table(writable, file=paste0(outdir, "SNP_aggregate_transcript_",cell[i], "_SAMPLED_DOWN", s, ".txt"), quote=F,
              row.names=F, col.names=F, sep="\t") # change the name of the file
  
  system(paste0("sed -e 's/, /\t/g' < /rds/general/user/aa19618/home/HMAGMA_Protocol/annotation_files/SNP_aggregate_transcript_",cell[i], "_SAMPLED_DOWN", s, ".txt >",outdir, cell[i], "_SAMPLED_DOWN", s, ".transcript.annot"))
  }
}
```



## HMAGMA output analysis for all regions 
```{r - HMAGMA output analysis }
load(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/gene_information.rda")
load(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda")

type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

sample<-1:10
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

#Directories to save the files 
dir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/HMAGMA_output_v3/sampled_down/"
outdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/"

#WITHOUT ASTROCYTES
#genedir
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_sd_without_astro_v4/")
genedir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_sd_without_astro_v4/"
#z stat directory 
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_sd_without_astro_v4/")
zdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_sd_without_astro_v4/"
#z sum stat diretory 
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_sd_without_astro_v4/")
zdir_sum<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_sd_without_astro_v4/"

##WITH ASTROCYTES
#genedir
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_sd_v3/")
#genedir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_sd_v3/"
##z stat directory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_sd_v3/")
#z sum stat diretory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_sd_v3/")
#zdir_sum<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_sd_v3/"
###zdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_sd_v3/"

#WITH ASTROCYTES ALL GENES
#genedir
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/gene_count_all_sd_v3/")
#genedir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/gene_count_all_sd_v3/"
#z stat directory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_all_sd_v3/")
#zdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_all_sd_v3/"
#z sum stat diretory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_all_sd_v3/")
#zdir_sum<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_all_sd_v3/"

#INTRONIC + EXONIC

dat<-c()
for(t in 1:length(type)) {
for(s in 1:length(sample)) {
for(c in 1:length(cell)) {
  diseasemat<-read.table(paste0(dir, cell[c], "_", sample[s], "_", type[t], "_gc41.genes.out"), header = T)
  
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
  
  diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
  
  #gene count per cell type per disease
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_", sample[s],"_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count) 
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  write.table(x = gene_count, file = paste0(genedir, type[t], "_", cell[c], "_", sample[s],"_count.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  #z mean per cell type per disease
  z_mean<-data.frame(sum(diseasemat_names$gene_zstat)/gene_count$nrow.diseasemat_names.) #•
  rownames(z_mean)<-paste0(type[t], "_", cell[c], "_", sample[s],"_z_mean") #•
  z_mean <- data.frame(names = row.names(z_mean), z_mean) 
  z_mean<- z_mean %>% data.frame(do.call("rbind", strsplit(as.character(z_mean$names), "_", fixed = TRUE)))
  write.table(x = z_mean, file = paste0(zdir, type[t], "_", cell[c], "_", sample[s],"_z_mean.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z sum per cell type per disease 
  z_sum<-data.frame(sum(diseasemat_names$gene_zstat))
  rownames(z_sum)<-paste0(type[t], "_", cell[c], "_", sample[s],"_z_sum") #•
  z_sum <- data.frame(names = row.names(z_sum), z_sum) 
  z_sum<- z_sum %>% data.frame(do.call("rbind", strsplit(as.character(z_sum$names), "_", fixed = TRUE)))
  write.table(x = z_sum, file = paste0(zdir_sum, type[t], "_", cell[c], "_", sample[s],"_z_sum.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #assign(paste0(type[t], "_", cell[c], "_", sample[s],"_count"), gene_count) 
  assign(paste0(type[t], "_", cell[c], "_", sample[s]), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_",  type[t], "_", sample[s], "_fdr005_output_genes_v4.rda"))
}
}
}

```


## HMAGMA output analysis for only 1) intronic + intergenic snps, 2) exonic + promoter snps
```{r - HMAGMA output analysis }

load(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/gene_information.rda")
load(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda")

type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")
sample<-1:10
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
region<-c("interactome_intronic", "pro_exon")

#Directories to save the files 
dir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/HMAGMA_output_v3/sampled_down/"
outdir<-"/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/"

#WITHOUT ASTROCYTES
#genedir
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_interactome_intronic_sd_without_astro_v4/")
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_pro_exon_sd_without_astro_v4/")
#z stat directory 
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_interactome_intronic_sd_without_astro_v4/")
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_pro_exon_sd_without_astro_v4/")
#z sum stat diretory 
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_interactome_intronic_sd_without_astro_v4/")
dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_pro_exon_sd_without_astro_v4/")

#WITH ASTROCYTES
#genedir
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/gene_count_interactome_intronic_sd_v3/")
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/gene_count_pro_exon_sd_v3/")
##z stat directory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_interactome_intronic_sd_v3/")
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_pro_exon_sd_v3/")
#z sum stat diretory 
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_interactome_intronic_sd_v3/")
#dir.create(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_pro_exon_sd_v3/")

dat<-c()
for(t in 1:length(type)) {
for(s in 1:10) {
for(c in 1:length(cell)) {
for (r in 1:length(region)){
  diseasemat<-read.table(paste0(dir, cell[c], "_", region[r], "_", s, "_", type[t], "_gc41.genes.out"), header = T)
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
  
   diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
   
  #gene count per cell type per disease
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_", region[r], "_", s,"_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count)
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  
  genedir<-paste0("~/HMAGMA_Protocol/output_annotation_v4/gene_count_", region[r], "_sd_without_astro_v4/")
  #genedir<-paste0("~/HMAGMA_Protocol/output_annotation_v3/gene_count_", region[r], "_sd_v3/")

  write.table(x = gene_count, file = paste0(genedir, type[t], "_", cell[c], "_", region[r], "_", s,"_count.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z mean per cell type per disease
  z_mean<-data.frame(sum(diseasemat_names$gene_zstat)/gene_count$nrow.diseasemat_names.) #•
  rownames(z_mean) <- paste0(type[t], "_", cell[c], "_", region[r], "_", s,"_z_mean")
  z_mean <- data.frame(names = row.names(z_mean), z_mean) 
  z_mean<- z_mean %>% data.frame(do.call("rbind", strsplit(as.character(z_mean$names), "_", fixed = TRUE)))
  
  zdir<-paste0("~/HMAGMA_Protocol/output_annotation_v4/z_mean_", region[r], "_sd_without_astro_v4/")
  #zdir<-paste0("~/HMAGMA_Protocol/output_annotation_v3/z_mean_", region[r], "_sd_v3/")
  write.table(x = z_mean, file = paste0(zdir, type[t], "_", cell[c], "_", region[r], "_", s,"_z_mean.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z sum per cell type per disease 
  z_sum<-data.frame(sum(diseasemat_names$gene_zstat))
  rownames(z_sum) <- paste0(type[t], "_", cell[c], "_", region[r], "_", s,"_z_sum")
  z_sum <- data.frame(names = row.names(z_sum), z_sum) 
  z_sum<- z_sum %>% data.frame(do.call("rbind", strsplit(as.character(z_sum$names), "_", fixed = TRUE)))
  
  zdir_sum<-paste0("~/HMAGMA_Protocol/output_annotation_v4/z_sum_", region[r], "_sd_without_astro_v4/")
  #zdir_sum<-paste0("~/HMAGMA_Protocol/output_annotation_v3/z_sum_", region[r], "_sd_v3/")
  
  write.table(x = z_sum, file = paste0(zdir_sum, type[t], "_", cell[c], "_", region[r], "_", s,"_z_sum.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #assign(paste0(type[t], "_", cell[c], "_", s,"_count"), gene_count) 
  assign(paste0(type[t], "_", cell[c], "_", s), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_",  type[t], "_", region[r], "_", s, "_fdr005_output_genes_v4.rda"))
}
}
}
} 
```

## HMAGMA output barplot

```{r Venn Diagram}
library(VennDetail)
library(Vennerable)

m<-list(Microglia = AD_microglia$gene_ensg)
n<-list( Neurons = AD_neurons$gene_ensg)
o<-list(Oligo = AD_oligo$gene_ensg)
bind<-c(m,n,o)
V3a<-Venn(Sets = bind)

m_intr<-list(Microglia = AD_intr_microglia$gene_ensg)
n_intr<-list( Neurons = AD_intr_neurons$gene_ensg)
o_intr<-list(Oligo = AD_intr_oligo$gene_ensg)

bind<-c(m_intr,n_intr,o_intr)
V3a<-Venn(Sets = bind)

m_ex<-list(Microglia = AD_ex_microglia$gene_ensg)
n_ex<-list( Neurons = AD_ex_neurons$gene_ensg)
o_ex<-list(Oligo = AD_ex_oligo$gene_ensg)
bind<-c(m_ex,n_ex,o_ex)
V3a<-Venn(Sets = bind)

plot(V3a, show=list(SetLabels=TRUE))
plot(V3a, type="ChowRuskey", show=list(SetLabels=FALSE))


```

##Combining gene counts for original sampled down (intronic, intergeni +exonic,promoter) GENE COUNT
```{r - gene count per disease per cell type}
#Merging all the files into one dataframe 

file_list <- list.files(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_sd_without_astro_v4/", full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/gene_count_sd_without_astro_v4.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```

##Combining gene counts for original sampled down (intronic, intergeni +exonic,promoter) Z MEAN
```{r - gene count per disease per cell type}
#Merging all the files into one dataframe 

file_list <- list.files(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_sd_without_astro_v3/", full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_mean_sampled_down_without_astro_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```

##Combining gene counts for original sampled down (intronic, intergeni +exonic,promoter) Z SUM
```{r - gene count per disease per cell type}
#Merging all the files into one dataframe 

file_list <- list.files(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_sd_without_astro_v3/", full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_sum_sampled_down_without_astro_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```

##Combining gene counts for either intronic, intergenic or exonic, promoter sampled down  GENE COUNT 
```{r}
#Merging all the files into one dataframe 

region<-c("interactome_intronic", "pro_exon")

file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_interactome_intronic_sd_without_astro_v4/"), full.names = TRUE)
file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_pro_exon_sd_without_astro_v4/"), full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/gene_count_interactome_intronic_sd_without_astro_v4.txt",row.names = F, col.names = T, quote = F, sep = "\t")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/gene_count_pro_exon_sd_without_astro_v4.txt",row.names = F, col.names = T, quote = F, sep = "\t")

split_dataset<- split(x = dataset, f = dataset$type)
```

##Combining gene counts for either intronic, intergenic or exonic, promoter sampled down  Z MEAN
```{r}
#Merging all the files into one dataframe 

region<-c("interactome_intronic", "pro_exon")

file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_interactome_intronic_sd_v3/"), full.names = TRUE)
file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_pro_exon_sd_v3/"), full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_mean_interactome_intronic_sd_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_mean_pro_exon_sd_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```

##Combining gene counts for either intronic, intergenic or exonic, promoter sampled down  Z SUM
```{r}
#Merging all the files into one dataframe 

region<-c("interactome_intronic", "pro_exon")

file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_interactome_intronic_sd_v3/"), full.names = TRUE)
file_list <- list.files(path = paste("/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_sum_pro_exon_sd_v3/"), full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "sample", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_sum_interactome_intronic_sd_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/z_sum_pro_exon_sd_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```



