## HMAGMA output analysis for all regions 
```{r - HMAGMA output analysis }

#Loading gene information and reference genome snps
load(file = "/required_files/gene_information.rda") #genes
load(file = "/required_files/snps_g1000.rda") #snps

#Cell names and disease names 
#type<-c("AD", "MS", "ALS", "PD", "SCZ")
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

#Directories to save the files 
dir<-"/HMAGMA_output_v3/original/"
outdir<-"/output_annotation_v4/"

#genedir
dir.create(path = "/output_annotation_v4/gene_count_v4/")
genedir<-"/output_annotation_v4/gene_count_v4/"

dat<-c()
for(t in 1:length(type)) {
for(c in 1:length(cell)) {
  
  #Loading HMAGMA output for each cell type and disease 
  diseasemat<-read.table(paste0(dir, cell[c], "_", type[t], "_gc41.genes.out"), header = T)
  
  #Calculating FDR (false discovery rate) of the gene p-value
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  #Adding gene names to ensgs for genes with FDR p<0.05
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  
  #Reformatting and adding column names 
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
  #Ordering genes by FDR p-value 
  diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
  
  #Adding a column with cell type and disease type
  try(diseasemat_names$cell<-paste0(cell[c]))
  try(diseasemat_names$disease<-paste0(type[t]))
  
  #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count) 
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  #write.table(x = gene_count, file = paste0(genedir, type[t], "_", cell[c],"_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")
  
  
  #Assigning a unique name to each file for each cell type for each disease

  assign(paste0(type[t], "_", cell[c]), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))


}
  }
  
data_list <- list(AD_Jansen2019_Microglia_interactome = AD_Jansen2019_Microglia_interactome, AD_Jansen2019_Neuronal_interactome = AD_Jansen2019_Neuronal_interactome, AD_Jansen2019_Oligo_interactome = AD_Jansen2019_Oligo_interactome, AD_Kunkle2019_Microglia_interactome=AD_Kunkle2019_Microglia_interactome, 
AD_Kunkle2019_Neuronal_interactome=AD_Kunkle2019_Neuronal_interactome,
AD_Kunkle2019_Oligo_interactome=AD_Kunkle2019_Oligo_interactome, 
PD_Nalls2019proxy_Microglia_interactome=PD_Nalls2019proxy_Microglia_interactome,
PD_Nalls2019proxy_Neuronal_interactome=PD_Nalls2019proxy_Neuronal_interactome,
PD_Nalls2019proxy_Oligo_interactome=PD_Nalls2019proxy_Oligo_interactome,
MS_Andlauer2016_Microglia_interactome=MS_Andlauer2016_Microglia_interactome,
MS_Andlauer2016_Neuronal_interactome=MS_Andlauer2016_Neuronal_interactome,
MS_Andlauer2016_Oligo_interactome=MS_Andlauer2016_Oligo_interactome, 
ALS_Rheenen2021_Microglia_interactome=ALS_Rheenen2021_Microglia_interactome,
ALS_Rheenen2021_Neuronal_interactome=ALS_Rheenen2021_Neuronal_interactome,
ALS_Rheenen2021_Oligo_interactome=ALS_Rheenen2021_Oligo_interactome,
SCZ_Trubetskoy2022_Microglia_interactome=SCZ_Trubetskoy2022_Microglia_interactome,
SCZ_Trubetskoy2022_Neuronal_interactome=SCZ_Trubetskoy2022_Neuronal_interactome,
SCZ_Trubetskoy2022_Oligo_interactome=SCZ_Trubetskoy2022_Oligo_interactome)


library(writexl)
write_xlsx(data_list, path = "/results/tables/hmagma_fdr005_output_genes_v4.xlsx")
  #Attaching rsids
  
  #Loading HMAGMA output snps for each gene for each cell type and disease 
  hmagma_gene_rsid<-read.table(file = paste0(dir, cell[c], "_", type[t], "_gc41.snp.index"),
                             skip = 3, header = T, sep = "\t")
  #Adding rsids to each gene 
  diseasemat_rsids<-merge(x = diseasemat_names, y = hmagma_gene_rsid, by.x="gene_ensg", by.y="GENE_NAME")
  diseasemat_rsids<-merge(x=diseasemat_rsids, y=snps, by.x="SNP_ID", by.y="rsid")
  
  #Keeping only snps that mapped to genes
  diseasemat_rsids<- diseasemat_rsids[diseasemat_rsids$STATUS_CODE > 0,]
  
  diseasemat_rsids<-diseasemat_rsids[,c(2,3,4,5,6,7,8,9,10,11,12,1,15,16)] %>% unique()
  colnames(diseasemat_rsids)<-c(c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps", "gene_zstat", "gene_p", "gene_fdr", "gene_biotype", "snp_rsid", "snp_chr", "snp_position"))
  save(diseasemat_rsids, file= paste0(outdir, cell[c], "_", type[t], "_fdr005_output_snp_v3.rda")) #Change name depending on FDR
}
#}
}

```


## HMAGMA output analysis for 1)intronic + intergenic, 2) prmoter + exonic regions 
```{r - HMAGMA output analysis }

#Loading gene information and reference genome snps
load(file = "/required_files/gene_information.rda")
load(file = "/required_files/snps_g1000.rda")

#Cell names, disease names and region types 
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
region<-c("intronic_intergenic", "pro_exon")

#Directories to save the files 
dir<-"/HMAGMA_output_v3/original/"
outdir<-"/output_annotation_v4/"

dir.create(path = "/output_annotation_v4/gene_count_intronic_intergenic_v4/")
dir.create(path = "/output_annotation_v4/gene_count_pro_exon_v4/")

#INTRONIC + INTERGENIC OR EXONIC + PROMOTER

dat<-c()
for(t in 1:length(type)) {
  for(c in 1:length(cell)) {
    for(r in 1:length(region)){
  
  #Loading HMAGMA output for each cell type and disease 
  diseasemat<-read.table(paste0(dir, cell[c], "_", region[r], "_", type[t], "_gc41.genes.out"), header = T)
  
  #Calculating FDR (false discovery rate) of the gene p-value
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  #Adding gene names to ensgs for genes with FDR p<0.05
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  
  #Reformatting and adding column names 
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
 
  #Ordering genes by FDR p-value
  diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
  
  #Adding a column with cell type and disease type
  try(diseasemat_names$cell<-paste0(cell[c]))
  try(diseasemat_names$disease<-paste0(type[t]))

  #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count) 
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  genedir<-paste0("~/HMAGMA_Protocol/output_annotation_v4/gene_count_", region[r], "_v4/")
  write.table(x = gene_count, file = paste0(genedir, type[t], "_", region[r], "_", cell[c], "_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")
  
  #Assigning a unique name to each file for each cell type for each disease
  assign(paste0(type[t], "_", cell[c], "_", region[r]), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_", region[r], "_" , type[t], "_fdr005_output_genes_v4.rda"))
    }
  }
}
```
