---
title: "H-MAGMA PIPELINE ON ORIGINAL PLAC"
author: "Aydan Askarova"
date: "25/01/2023"
output: html_document
---

## Packages

```{r packages}
options(stingAsFactors=F)
library(GenomicRanges)
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(reshape)
library(readxl)

library(openxlsx)
```

## Required files

```{r files}
load ("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/exon_promoranges.rda") #exonranges, promoterranges
load ("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda") #snps
load("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/snp_locating_in_exon_promoter_transcript_level_g1000.rda") #snpro, snpexon
load("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/non_exonic_promoter_snp_g1000.rda") #snpranges
```

## PLAC-seq filtered to enhancer promoter interactions 
```{r loading PLAC-seq}
#Cell names
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")

#Directories to save the files 
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/"
inputdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/required_files/H3K27AC_peaks/"

#Looping
dat <- c()
for(i in 1:length(cell)) { 
  
    #Reading the plac-seq files for each cell type
    hic <- read_excel(path = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/PLAC/aay0793-nott-table-s5.xlsx", sheet = paste0(cell[i]), skip = 2, col_names = T)
    
    #Doubling the hic and flipping the start1-end1 with int1-int2 (this is for GRanges function and IRanges)
    hic.int1 <- hic[,1:6]
    hic.int2 <- hic[,c(4:6,1:3)]
    colnames(hic.int1) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    colnames(hic.int2) <- c("chrom1", "start1", "end1", "chrom2", "start2", "end2")
    hic <- rbind(hic.int1, hic.int2)
    
    #Creating GRanges for findOverlaps() function
    hicranges <- GRanges(hic$chrom1, IRanges(as.numeric(hic$start1), as.numeric(hic$end1)),
                        int1=hic$start2,int2=hic$end2)
    
    #Reading promoters data (Nott et al., 2019 paper)
    promoters<-read.table(file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/", cell[i],   "_promoters.bed"), header=F, sep="\t")
    colnames(promoters)<-c("chr", "start", "end")
    promoters_ranges<-GRanges(promoters$chr, IRanges(as.numeric(promoters$start), as.numeric(promoters$end)))
  
    #Selecting plac-seq interactions that overlap with promoters from Chip-seq (this is just in case the interactions pulled by plac-seq are not tied to promoters)
    olap<-findOverlaps(query = hicranges, subject = promoters_ranges) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    placranges<-hicranges[queryHits(olap)]
    mcols(placranges)<-cbind(mcols(hicranges[queryHits(olap)]), mcols(promoters_ranges[subjectHits(olap)])) 

    #Selecing filtered plac-seq interactions overlapping with genomic exons
    olap <- findOverlaps(query = placranges,subject = exonranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    exonint <- placranges[queryHits(olap)];
    mcols(exonint) <- cbind(mcols(placranges[queryHits(olap)]), mcols(exonranges[subjectHits(olap)]))

    #Selecing filtered plac-seq interactions overlapping with genomic promoters (1.5kb upstream and 500bp downstream of TSS)
    olap <- findOverlaps(query = placranges, subject = promoterranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    proint <- placranges[queryHits(olap)];
    mcols(proint) <- cbind(mcols(placranges[queryHits(olap)]), mcols(promoterranges[subjectHits(olap)]))

    #Combining filtered plac-seq interactions containing exons and promoters 
    generanges <- c(exonint,proint)
    
    #Making a dataframe with plac-ranges interacting regions and ONLY THEN removing row duplicates (since some plac interactions will overlap between exons and promoters)
    genebed <- data.frame(chr=seqnames(generanges), snp.start=generanges$int1, snp.end=generanges$int2,
                          gene.start=start(generanges), gene.end=start(generanges)+width(generanges)-1,
                          ensg=generanges$gene)
    genebed <- unique(genebed) 
    
    #Making GRanges of the combined filtered plac-seq interactions and making IRanges exon and promoter interacting regions
    genesnpranges <- GRanges(genebed$chr, IRanges(genebed$snp.start, genebed$snp.end), snp.start=genebed$snp.start, snp.end=genebed$snp.end, gene.start=genebed$gene.start, gene.end=genebed$gene.end, ensg=genebed$ensg)

    #Selecting SNPs that fall within regions that interact with exons and promoters (non_exonic_promoter_snp_g1000.rda = snpranges)
    olap <- findOverlaps(query = snpranges, subject = genesnpranges); #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    snpint <- snpranges[queryHits(olap)];
    mcols(snpint) <- cbind(mcols(snpranges[queryHits(olap)]), mcols(genesnpranges[subjectHits(olap)]))
    
    #Making a dataframe with plac-ranges interacting regions and ONLY THEN removing row duplicates 
    snpint_data<-as.data.frame(snpint) #•
    snpint_data <- unique(snpint_data) #•
    
    #Making GRanges for the SNPs located within exon/promoter interacting regions 
    snpint<-GRanges(seqnames = snpint_data$seqnames, ranges = IRanges(snpint_data$start, snpint_data$end), strand = snpint_data$strand,
                    rsid=snpint_data$rsid, ensg=snpint_data$ensg, snp.start=snpint_data$snp.start, snp.end=snpint_data$snp.end, gene.start=snpint_data$gene.start, gene.end=snpint_data$gene.end)
    
    save(snpint, file=paste0(outdir, "Hi-C_transcript_interacting_snp_",cell[i], ".rda"))
    
    #Reading enhancers data  (Nott et al., 2019 paper) 
    enhancers<-read.table(file=paste0("/Volumes/aa19618/home/HMAGMA_Protocol/required_files/enhancers_promoters/", cell[i], "_enhancers.bed"), header=F, sep="\t")
    colnames(enhancers)<-c("chr", "start", "end")
   enhancers_ranges<-GRanges(enhancers$chr, IRanges(as.numeric(enhancers$start), as.numeric(enhancers$end)))   
  
    #Selecting exonic/promoter interacting SNPS overlapping with enhancers   
    olap<-findOverlaps(query = snpint, subject = enhancers_ranges) #default maxgap=-1L, when one range's start/end strictly inside the other, the gap is considered to be -1.
    enhancer_snp_ranges<-snpint[queryHits(olap)]
    mcols(enhancer_snp_ranges)<-cbind(mcols(snpint[queryHits(olap)]), mcols(enhancers_ranges[subjectHits(olap)]))

    #Making a dataframe with exonic/promoter interacting enhancer SNPs and ONLY THEN removing row duplicates 
    enhancer_snp_ranges_data<-as.data.frame(enhancer_snp_ranges) #•
    enhancer_snp_ranges_data<-unique(enhancer_snp_ranges_data) #• 
  
    #Making GRanges for the enhancer SNPs located within exon/promoter interacting regions 
    enhancer_snp_ranges<-GRanges(seqnames = enhancer_snp_ranges_data$seqnames, ranges = IRanges(enhancer_snp_ranges_data$start, enhancer_snp_ranges_data$end), strand = enhancer_snp_ranges_data$strand,
                    rsid=enhancer_snp_ranges_data$rsid, ensg=enhancer_snp_ranges_data$ensg)
    
    save(enhancer_snp_ranges, file=paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_",cell[i], ".rda"))
}

```

## Creating annotation file - PROMOTER + EXONIC + INTRONIC + INTERGENIC regions

```{r}
#Annotation file - exo, pro, intr, inter

#Looping
dat<-c()
for(i in 1:length(cell)){
  
  #Loading GRanges for the enhancer SNPs located within the exonic/promoter interacting regions and making into dataframe 
  load(file=paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_",cell[i],".rda"))
  snpdat <- data.frame(chr=seqnames(enhancer_snp_ranges), bp=start(enhancer_snp_ranges), rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg)
  
  #Saving exonic/promoter SNPs and their genes and combining into one dataframe (snpgene)
  snpromat <- unique(data.frame(rsid=snpro$rsid, ensg=snpro$gene))
  snpexonmat <- unique(data.frame(rsid=snpexon$rsid, ensg=snpexon$gene))
  snpgene<-unique(rbind(snpromat,snpexonmat))
  
  #Saving exonic/promoter interacting enhancer SNPs and associated genes 
  snpintmat<-unique(data.frame(rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg))
  
  #Selecting genes and all the associated SNPs that have interactions with enhancers 
  
  #1. Selecting all the genes with enhancer interactions, their enhancer SNPs, their exonic/promoter SNPs
  snpcomb_loop<-merge(x = snpgene, y = snpintmat, by.x="ensg", by.y="ensg", all.y=TRUE) #including 
  #2. Combining columns 2 and 3 with SNPs from enhancer SNPs and exonic/promoter SNPs
  snpcomb_loop_melt<-melt(snpcomb_loop, id=c("ensg"))
  #3. Making rsid the 1st column and ensg 3rd column
  snpcomb_loop_melt<-snpcomb_loop_melt[,c(3,1)]
  #4. Removing any duplicate rows
  snpcomb_loop_melt<-unique(snpcomb_loop_melt)
  #5. Omitting any NAs in rsid column 
  snpcomb_loop_melt<-na.omit(snpcomb_loop_melt)
  #6. Naming the columns 
  colnames(snpcomb_loop_melt)<-c("rsid", "ensg")
  #7. Saving the dataframe 
  save(snpcomb_loop_melt, file=paste0(outdir,"SNP_to_transcript_comb_", cell[i], ".rda")) 
  
  #Making MAGMA compatible annotation file 
  
  #1. Aggregating all the SNPs associated with the same gene into one row
  snpagg <- aggregate(snpcomb_loop_melt, list(snpcomb_loop_melt$ensg), unique)
  #2. Loading gencode41_37 for promoter, exon and gene coordinates 
  load(file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/gencode41_37_pro_exo_gene.rda")
  colnames(gene) <- c("chr", "start", "end", "ensg")
  #3. Removing "chr" from the first column 
  gene<-transform(gene, chr=as.character(chr))
  gene <- gene[grep("chr", gene$chr),]
  gene$chr <- unlist(lapply(strsplit(gene$chr, "chr"), '[[', 2))
  #4. Creating gene coordinate infomration (chr:start:end)
  gene$index <- paste(gene$chr, gene$start, gene$end, sep=":")
  #5. Adding the index information to the gene-snp file (snpagg)
  snpagg$index = gene[match(snpagg$ensg, gene$ensg),"index"]
  #6. Removing any NA
  snpagg <- snpagg[!is.na(snpagg$index),]
  snpaggconv <- snpagg[,c("ensg", "index", "rsid")]
  #7. Formatting and saving the table
  writable <- format(snpaggconv)
  write.table(writable, file=paste0(outdir, "SNP_aggregate_transcript_",cell[i],".txt"), quote=F,
              row.names=F, col.names=F, sep="\t") # change the name of the file
  #8. Writing it in a list format, each gene and associated information per line 
  system(paste0("sed -e 's/, /\t/g' < /Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/SNP_aggregate_transcript_",cell[i],".txt >",outdir, cell[i],".transcript.annot"))
}
```

## Creating annotation file - PROMOTER + EXONIC regions

```{r}
#Annotation file - exo, pro

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/"

dat<-c()
for(i in 1:length(cell)){
  
  #Loading GRanges for the enhancer SNPs located within the exonic/promoter interacting regions and making into dataframe 
  load(file = paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_", cell[i], ".rda"))
  snpdat <- data.frame(chr=seqnames(enhancer_snp_ranges), bp=start(enhancer_snp_ranges), rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg)
  
  #Saving exonic/promoter SNPs and their genes and combining into one dataframe (snpgene)
  snpromat <- unique(data.frame(rsid=snpro$rsid, ensg=snpro$gene))
  snpexonmat <- unique(data.frame(rsid=snpexon$rsid, ensg=snpexon$gene))
  snpgene<-unique(rbind(snpromat,snpexonmat))
  
  #Saving exonic/promoter interacting enhancer SNPs and associated genes 
  snpintmat<-unique(data.frame(rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg))
  
  #Selecting genes and their exonic/promoter SNPs based on whether they have interactions with enhancers 
  
  #1. Only keeping genes that have enhacer SNPs
  snpintmat_ensg_only<-as.data.frame(snpintmat[2])
  #2. Selecting only those exonic/promoter genes that also have enhancer SNPs
  snpcomb_loop_melt<-merge(x = snpgene, y = snpintmat_ensg_only, by.x="ensg", by.y="ensg")
  #3. Making rsid the 1st column and ensg 3rd column
  snpcomb_loop_melt<-snpcomb_loop_melt[,c(2,1)]
  #4. Removing any duplicate rows
  snpcomb_loop_melt<-unique(snpcomb_loop_melt)
  
  #Making MAGMA compatible annotation file 
  
  #1. Aggregating all the SNPs associated with the same gene into one row
  snpagg <- aggregate(snpcomb_loop_melt, list(snpcomb_loop_melt$ensg), unique)
  #2. Loading gencode41_37 for promoter, exon and gene coordinates 
  load(file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/gencode41_37_pro_exo_gene.rda")
  colnames(gene) <- c("chr", "start", "end", "ensg")
  #3. Removing "chr" from the first column 
  gene<-transform(gene, chr=as.character(chr))
  gene <- gene[grep("chr", gene$chr),]
  gene$chr <- unlist(lapply(strsplit(gene$chr, "chr"), '[[', 2))
  #4. Creating gene coordinate infomration (chr:start:end)
  gene$index <- paste(gene$chr, gene$start, gene$end, sep=":")
  #5. Adding the index information to the gene-snp file (snpagg)
  snpagg$index = gene[match(snpagg$ensg, gene$ensg),"index"]
  #6. Removing any NA
  snpagg <- snpagg[!is.na(snpagg$index),]
  snpaggconv <- snpagg[,c("ensg", "index", "rsid")]
  #7. Formatting and saving the table
  writable <- format(snpaggconv)
  write.table(writable, file=paste0(outdir, "SNP_aggregate_transcript_pro_exon_",cell[i], ".txt"), quote=F,
              row.names=F, col.names=F, sep="\t") # change the name of the file
  #8. Writing it in a list format, each gene and associated information per line 
  system(paste0("sed -e 's/, /\t/g' < /Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/SNP_aggregate_transcript_pro_exon_",cell[i],".txt >",outdir, cell[i],"_pro_exon.transcript.annot"))
}
```

## Creating annotation file - INTRONIC + INTERGENIC regions

```{r}
#Annotation file intronic and intergenic 

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/"

dat<-c()
for(i in 1:length(cell)){
  
  #Loading GRanges for the enhancer SNPs located within the exonic/promoter interacting regions and making into dataframe 
  load(file = paste0(outdir, "Hi-C_transcript_interacting_snp_H3K27ac_", cell[i], ".rda"))
  snpdat <- data.frame(chr=seqnames(enhancer_snp_ranges), bp=start(enhancer_snp_ranges), rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg)
  
  #Saving exonic/promoter interacting enhancer SNPs and associated genes 
  snpintmat<-unique(data.frame(rsid=enhancer_snp_ranges$rsid, ensg=enhancer_snp_ranges$ensg))
  snpcomb_loop_melt<-snpintmat
  
  #Making MAGMA compatible annotation file 
  
  #1. Aggregating all the SNPs associated with the same gene into one row
  snpagg <- aggregate(snpcomb_loop_melt, list(snpcomb_loop_melt$ensg), unique)
  #2. Loading gencode41_37 for promoter, exon and gene coordinates 
  load(file="/Volumes/aa19618/home/HMAGMA_Protocol/required_files/gencode41_37_pro_exo_gene.rda")
  colnames(gene) <- c("chr", "start", "end", "ensg")
  #3. Removing "chr" from the first column
  gene<-transform(gene, chr=as.character(chr))
  gene <- gene[grep("chr", gene$chr),]
  gene$chr <- unlist(lapply(strsplit(gene$chr, "chr"), '[[', 2))
  #4. Creating gene coordinate infomration (chr:start:end)
  gene$index <- paste(gene$chr, gene$start, gene$end, sep=":")
  #5. Adding the index information to the gene-snp file (snpagg)
  snpagg$index = gene[match(snpagg$ensg, gene$ensg),"index"]
  #6. Removing any NA
  snpagg <- snpagg[!is.na(snpagg$index),]
  snpaggconv <- snpagg[,c("ensg", "index", "rsid")]
  #7. Formatting and saving the table 
  writable <- format(snpaggconv)
  
  write.table(writable, file=paste0(outdir, "SNP_aggregate_transcript_intronic_intergenic_",cell[i],".txt"), quote=F,
              row.names=F, col.names=F, sep="\t") # change the name of the file
  #8. Writing it in a list format, each gene and associated information per line 
  system(paste0("sed -e 's/, /\t/g' < /Volumes/aa19618/home/HMAGMA_Protocol/annotation_files/SNP_aggregate_transcript_intronic_intergenic_",cell[i],".txt >",outdir, cell[i],"_intronic_intergenic.transcript.annot"))
}

```


## HMAGMA output analysis for all regions 
```{r - HMAGMA output analysis }

#Loading gene information and reference genome snps
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/gene_information.rda") #genes
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda") #snps

#Cell names and disease names 
#type<-c("AD", "MS", "ALS", "PD", "SCZ")
cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

#Directories to save the files 
dir<-"/Volumes/aa19618/home/HMAGMA_Protocol/HMAGMA_output_v3/original/"
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/"

#WITH ASTROCYTES
#genedir
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_v4/")
genedir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_v4/"
#z stat directory 
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_v4/")
zdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_v4/"
#z sum stat diretory 
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_v4/")
zdir_sum<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_v4/"

#INTRONIC + EXONIC

#Looping 
dat<-c()
for(t in 1:length(type)) {
for(c in 1:length(cell)) {
  
  #Loading HMAGMA output for each cell type and disease 
  diseasemat<-read.table(paste0(dir, cell[c], "_", type[t], "_gc41.genes.out"), header = T)
  
  #Calculating FDR (false discovery rate) of the gene p-value
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  #Adding gene names to ensgs for genes with FDR p<0.05
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  
  #Reformatting and adding column names 
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
  #Ordering genes by FDR p-value 
  diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
  
  #Adding a column with cell type and disease type
  try(diseasemat_names$cell<-paste0(cell[c]))
  try(diseasemat_names$disease<-paste0(type[t]))
  
  #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count) 
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  #write.table(x = gene_count, file = paste0(genedir, type[t], "_", cell[c],"_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z mean per cell type per disease
  z_mean<-data.frame(sum(diseasemat_names$gene_zstat)/gene_count$nrow.diseasemat_names.) #•
  rownames(z_mean)<-paste0(type[t], "_", cell[c], "_z_mean") #•
  z_mean <- data.frame(names = row.names(z_mean), z_mean) 
  z_mean<- z_mean %>% data.frame(do.call("rbind", strsplit(as.character(z_mean$names), "_", fixed = TRUE)))
  #write.table(x = z_mean, file = paste0(zdir, type[t], "_", cell[c], "_z_mean.txt"), 
   #           row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z sum per cell type per disease 
  z_sum<-data.frame(sum(diseasemat_names$gene_zstat))
  rownames(z_sum)<-paste0(type[t], "_", cell[c], "_z_sum") #•
  z_sum <- data.frame(names = row.names(z_sum), z_sum) 
  z_sum<- z_sum %>% data.frame(do.call("rbind", strsplit(as.character(z_sum$names), "_", fixed = TRUE)))
  #write.table(x = z_sum, file = paste0(zdir_sum, type[t], "_", cell[c], "_z_sum.txt"), 
 #             row.names = F, col.names = F, quote = F, sep = "\t")
  
  #Assigning a unique name to each file for each cell type for each disease
  #assign(paste0(type[t], "_", cell[c],"_count"), gene_count) 
  assign(paste0(type[t], "_", cell[c]), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))


}
  }
  
data_list <- list(AD_Jansen2019_Microglia_interactome = AD_Jansen2019_Microglia_interactome, AD_Jansen2019_Neuronal_interactome = AD_Jansen2019_Neuronal_interactome, AD_Jansen2019_Oligo_interactome = AD_Jansen2019_Oligo_interactome, AD_Kunkle2019_Microglia_interactome=AD_Kunkle2019_Microglia_interactome, 
AD_Kunkle2019_Neuronal_interactome=AD_Kunkle2019_Neuronal_interactome,
AD_Kunkle2019_Oligo_interactome=AD_Kunkle2019_Oligo_interactome, 
PD_Nalls2019proxy_Microglia_interactome=PD_Nalls2019proxy_Microglia_interactome,
PD_Nalls2019proxy_Neuronal_interactome=PD_Nalls2019proxy_Neuronal_interactome,
PD_Nalls2019proxy_Oligo_interactome=PD_Nalls2019proxy_Oligo_interactome,
MS_Andlauer2016_Microglia_interactome=MS_Andlauer2016_Microglia_interactome,
MS_Andlauer2016_Neuronal_interactome=MS_Andlauer2016_Neuronal_interactome,
MS_Andlauer2016_Oligo_interactome=MS_Andlauer2016_Oligo_interactome, 
ALS_Rheenen2021_Microglia_interactome=ALS_Rheenen2021_Microglia_interactome,
ALS_Rheenen2021_Neuronal_interactome=ALS_Rheenen2021_Neuronal_interactome,
ALS_Rheenen2021_Oligo_interactome=ALS_Rheenen2021_Oligo_interactome,
SCZ_Trubetskoy2022_Microglia_interactome=SCZ_Trubetskoy2022_Microglia_interactome,
SCZ_Trubetskoy2022_Neuronal_interactome=SCZ_Trubetskoy2022_Neuronal_interactome,
SCZ_Trubetskoy2022_Oligo_interactome=SCZ_Trubetskoy2022_Oligo_interactome)


library(writexl)
write_xlsx(data_list, path = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/hmagma_fdr005_output_genes_v4.xlsx")
  #Attaching rsids
  
  #Loading HMAGMA output snps for each gene for each cell type and disease 
  hmagma_gene_rsid<-read.table(file = paste0(dir, cell[c], "_", type[t], "_gc41.snp.index"),
                             skip = 3, header = T, sep = "\t")
  #Adding rsids to each gene 
  diseasemat_rsids<-merge(x = diseasemat_names, y = hmagma_gene_rsid, by.x="gene_ensg", by.y="GENE_NAME")
  diseasemat_rsids<-merge(x=diseasemat_rsids, y=snps, by.x="SNP_ID", by.y="rsid")
  
  #Keeping only snps that mapped to genes
  diseasemat_rsids<- diseasemat_rsids[diseasemat_rsids$STATUS_CODE > 0,]
  
  diseasemat_rsids<-diseasemat_rsids[,c(2,3,4,5,6,7,8,9,10,11,12,1,15,16)] %>% unique()
  colnames(diseasemat_rsids)<-c(c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps", "gene_zstat", "gene_p", "gene_fdr", "gene_biotype", "snp_rsid", "snp_chr", "snp_position"))
  save(diseasemat_rsids, file= paste0(outdir, cell[c], "_", type[t], "_fdr005_output_snp_v3.rda")) #Change name depending on FDR
}
#}
}

```


## HMAGMA output analysis for 1)intronic + intergenic, 2) prmoter + exonic regions 
```{r - HMAGMA output analysis }

#Loading gene information and reference genome snps
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/gene_information.rda")
load(file = "/Volumes/aa19618/home/HMAGMA_Protocol/required_files/snps_g1000.rda")

#Cell names, disease names and region types 
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022", "Longevity_Deelan201990th")

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
region<-c("intronic_intergenic", "pro_exon")

#Directories to save the files 
dir<-"/Volumes/aa19618/home/HMAGMA_Protocol/HMAGMA_output_v3/original/"
outdir<-"/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/"

dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_intronic_intergenic_v4/")
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_pro_exon_v4/")
#z stat directory 
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_intronic_intergenic_v4/")
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_mean_pro_exon_v4/")
#z sum stat diretory 
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_intronic_intergenic_v4/")
dir.create(path = "/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v4/z_sum_pro_exon_v4/")

#INTRONIC + INTERGENIC OR EXONIC + PROMOTER

#Looping 
dat<-c()
for(t in 1:length(type)) {
  for(c in 1:length(cell)) {
    for(r in 1:length(region)){
  
  #Loading HMAGMA output for each cell type and disease 
  diseasemat<-read.table(paste0(dir, cell[c], "_", region[r], "_", type[t], "_gc41.genes.out"), header = T)
  
  #Calculating FDR (false discovery rate) of the gene p-value
  diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
  
  #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
  diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
  diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
  diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
  
  #Adding gene names to ensgs for genes with FDR p<0.05
  diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
  
  #Reformatting and adding column names 
  diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18)]
  colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                   "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype") 
 
  #Ordering genes by FDR p-value
  diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
  
  #Adding a column with cell type and disease type
  try(diseasemat_names$cell<-paste0(cell[c]))
  try(diseasemat_names$disease<-paste0(type[t]))

  #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
  gene_count<-data.frame(nrow(diseasemat_names))
  rownames(gene_count) <- paste0(type[t], "_", cell[c], "_count")
  gene_count <- data.frame(names = row.names(gene_count), gene_count) 
  gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
  genedir<-paste0("~/HMAGMA_Protocol/output_annotation_v4/gene_count_", region[r], "_v4/")
  write.table(x = gene_count, file = paste0(genedir, type[t], "_", region[r], "_", cell[c], "_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z mean per cell type per disease
  z_mean<-data.frame(sum(diseasemat_names$gene_zstat)/gene_count$nrow.diseasemat_names.) #•
  rownames(z_mean)<-paste0(type[t], "_", cell[c], "_z_mean") #•
  z_mean <- data.frame(names = row.names(z_mean), z_mean) 
  z_mean<- z_mean %>% data.frame(do.call("rbind", strsplit(as.character(z_mean$names), "_", fixed = TRUE)))
  zdir<-paste0("~/HMAGMA_Protocol/output_annotation_v4/z_mean_", region[r], "_v4/")
  write.table(x = z_mean, file = paste0(zdir, type[t], "_", region[r], "_", cell[c], "_z_mean.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  #z sum per cell type per disease 
  z_sum<-data.frame(sum(diseasemat_names$gene_zstat))
  rownames(z_sum)<-paste0(type[t], "_", cell[c], "_z_sum") #•
  z_sum <- data.frame(names = row.names(z_sum), z_sum) 
  z_sum<- z_sum %>% data.frame(do.call("rbind", strsplit(as.character(z_sum$names), "_", fixed = TRUE)))
  zdir_sum<-paste0("~/HMAGMA_Protocol/output_annotation_v4/z_sum_", region[r], "_v4/")
  write.table(x = z_sum, file = paste0(zdir_sum, type[t], "_", region[r], "_", cell[c], "_z_sum.txt"), 
              row.names = F, col.names = F, quote = F, sep = "\t")
  
  
  #Assigning a unique name to each file for each cell type for each disease
  #assign(paste0(type[t], "_", cell[c],"_", region[r], "_count"), gene_count) 
  assign(paste0(type[t], "_", cell[c], "_", region[r]), diseasemat_names)
  
  save(diseasemat_names, file= paste0(outdir, cell[c], "_", region[r], "_" , type[t], "_fdr005_output_genes_v4.rda"))
    }
  }
}
  #Attaching rsids
  
  #Loading HMAGMA output snps for each gene for each cell type and disease 
  #hmagma_gene_rsid<-read.table(file = paste0(dir, cell[c], "_", region[r], "_", type[t], "_gc41.snp.index"),
   #                         skip = 3, header = T, sep = "\t")
  
  #Adding rsids to each gene 
  #diseasemat_rsids<-merge(x = diseasemat_names, y = hmagma_gene_rsid, by.x="gene_ensg", by.y="GENE_NAME")
  #diseasemat_rsids<-merge(x=diseasemat_rsids, y=snps, by.x="SNP_ID", by.y="rsid")
  
  #Keeping only snps that mapped to genes
  #diseasemat_rsids<- diseasemat_rsids[diseasemat_rsids$STATUS_CODE > 0,]
  
  #diseasemat_rsids<-diseasemat_rsids[,c(2,3,4,5,6,7,8,9,10,11,12,1,15,16)] %>% unique()
  #colnames(diseasemat_rsids)<-c(c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
  #                                 "gene_strand", "nsnps", "gene_zstat", "gene_p", "gene_fdr", "gene_biotype", "snp_rsid", #"snp_chr", "snp_position"))
#  save(diseasemat_rsids, file= paste0(outdir, cell[c], "_", region[r], "_", type[t], "_fdr005_output_snp.rda")) #Change name depending on FDR
#}
#}
#}
```

```{r}
#Merging all the files into one dataframe 

file_list <- list.files(path = paste("/Volumes/aa19618/home/HMAGMA_Protocol/output_annotation_v3/z_mean_pro_exon_v3/"), full.names = TRUE)

for (file in file_list){
      
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }

}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "cell", "interactome", "count")
write.table(x = dataset, file = "/Volumes/aa19618/home/HMAGMA_Protocol/results/tables/z_mean_pro_exon_v3.txt",row.names = F, col.names = T, quote = F, sep = "\t")
```