##MAGMA OUTPUT - FILTERING AND SELECTING RELEVANT GENES
#Loading gene information and reference genome snps
load(file = "~/HMAGMA_Protocol/required_files/gene_information.rda") #genes
load(file = "~/HMAGMA_Protocol/required_files/snps_g1000.rda") #snps

#Cell names and disease names 
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")

#Directories to save the files 
dir<-"~/HMAGMA_Protocol/HMAGMA_output_v3/original/"
outdir<-"~/HMAGMA_Protocol/output_annotation_v4/"

#genedir
dir.create(path = "~/HMAGMA_Protocol/output_annotation_v4/gene_count_og_magma/")
genedir<-"~/HMAGMA_Protocol/output_annotation_v4/gene_count_og_magma/"

#Extracting signigicant genes - AFTER MAGMA 
library(dplyr)
#Looping 
dat<-c()
for(t in 1:length(type)) {

    
    #Loading HMAGMA output for each cell type and disease 
    diseasemat<-read.table(paste0(dir, "og_magma_", type[t], "_gc41.genes.out"), header = T)

    #Calculating FDR (false discovery rate) of the gene p-value
    diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
    
    #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
    diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
    diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
    diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
    
    # Example ENSG IDs
    ensg_ids <- diseasemat_0.05$GENE
    
    # Get HGNC symbol and description
    gene_info <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol", "description"),
                       filters = "ensembl_gene_id",
                       values = ensg_ids,
                       mart = ensembl)
    
    print(gene_info)
    
    diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
    
    #Adding gene names to ensgs for genes with FDR p<0.05
    diseasemat_names<-merge(x =diseasemat_names, y = gene_info, by.x="GENE", by.y="ensembl_gene_id", all.x=TRUE)
    
    #Reformatting and adding column names 
    diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18,21)]
    colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                  "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype", "gene_description") 
    #Ordering genes by FDR p-value 
    diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
    
    #Adding a column with cell type and disease type
    try(diseasemat_names$disease<-paste0(type[t]))
    
    #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
    gene_count<-data.frame(nrow(diseasemat_names))
    rownames(gene_count) <- paste0(type[t], "_count")
    gene_count <- data.frame(names = row.names(gene_count), gene_count) 
    gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
    write.table(x = gene_count, file = paste0(genedir, type[t], "_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")

    save(diseasemat_names, file= paste0(outdir, type[t], "og_magma_fdr005_output_genes_v4.rda"))
    
    
  }


#Merging all the files into one dataframe 

file_list <- list.files(path = "/rds/general/user/aa19618/home/HMAGMA_Protocol/output_annotation_v4/gene_count_og_magma/", full.names = TRUE)

for (file in file_list){
  
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=FALSE, sep="\t")
  }
  
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=FALSE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }
  
}

dataset <- unique(dataset)
colnames(dataset) <- c("ID", "gene_count", "type", "gwas_type", "count")
write.table(x = dataset, file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/gene_count_og_magma.txt",row.names = F, col.names = T, quote = F, sep = "\t")

split_dataset<- split(x = dataset, f = dataset$type)


##Barplot of magma exonic+promoter 
library(ggplot2)
dataset<-read.table(file = "/rds/general/user/aa19618/home/HMAGMA_Protocol/results/tables/gene_count_og_magma.txt", header=T, sep="\t")
dataset<-dataset[!grepl("Longevity_Deelan201990th_", dataset$ID),]
dataset$gene_count_disc<-as.numeric(as.character(dataset$gene_count))
dataset$TYPE<-paste0(dataset$type, "_", dataset$gwas_type)


dataset$TYPE <- factor(dataset$TYPE, levels = c("AD_Jansen2019",  "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022"), ordered = TRUE)

# Define a custom transformation for the y-axis
custom_trans <- function(x) {
  ifelse(x <= 700, x, 700 + (x - 700) / 10) # Values above 100 are scaled down
}

inverse_trans <- function(x) {
  ifelse(x <= 700, x, 700 + (x - 700) * 10) # Inverse scaling for axis labels
}

# Custom transformation for the y-axis
custom_breaks <- c(seq(0, 700, 50), seq(700, 8000, 1000)) # Define breaks
custom_labels <- as.character(c(seq(0, 700, 50), seq(700, 8000, 1000))) # Corresponding labels
"darkorange2", "goldenrod1", "midnightblue","orangered4", "slateblue3", "olivedrab3"
# grouped boxplot
ggplot<-ggplot(dataset, aes(x=TYPE, y=gene_count, fill=TYPE)) + 
  geom_bar(position=position_dodge(width = 0.95), stat="identity") +
  xlab("Disease type") +
  ylab("Gene count") +
  theme(text = element_text(size=15)) +
  guides(fill=guide_legend(title="Cell type")) +
  scale_y_continuous(
    breaks = custom_breaks,
    labels = custom_labels,
    trans = scales::trans_new("custom", custom_trans, inverse_trans)
  ) +
  scale_fill_manual(values=c("darkorange2", "goldenrod1", "midnightblue","orangered4", "slateblue3", "olivedrab3")) +
#  facet_wrap(~TYPE, scale="free") +
  theme_classic() +
  theme(legend.position = "bottom",legend.direction = "horizontal", legend.title = element_blank(), axis.text.x=element_blank(), strip.background = element_blank()) + 
  ggtitle(label = "Gene count based on exonic and promoter SNPs v5")

ggsave(filename = "gene_count_og_magma_barplot.pdf", plot = last_plot(), device = "pdf", path = "~/HMAGMA_Protocol/results/graphs/barplots/",  width = 20, height = 15, units = "cm")
```

##overlapping magma genes wiht each cell type per disease
library(openxlsx)
outdir<-"~/HMAGMA_Protocol/output_annotation_v4/"

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
cell_name<-c("Microglia_interactome"="microglia", "Neuronal_interactome" = "neurons", "Oligo_interactome"="oligo")
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")
type_name<-c("AD_Jansen2019"="AD_Jansen2019", "AD_Kunkle2019"="AD_Kunkle2019", "PD_Nalls2019proxy"="PD", "MS_Andlauer2016"="MS", "ALS_Rheenen2021"="ALS", "SCZ_Trubetskoy2022"="SCZ")

wb <- createWorkbook()

for(t in 1:length(type)){
  load(file = paste0(paste0(outdir, type[t], "og_magma_fdr005_output_genes_v4.rda")))
       magma_genes<-diseasemat_names
       addWorksheet(wb = wb, sheetName = paste0(type_name[[type[t]]], "_magma"))
       writeData(wb, sheet = paste0(type_name[[type[t]]], "_magma"), magma_genes)
       
       for(c in 1:length(cell)){
         load(file=paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))
              hmagma_genes<-diseasemat_names 
              
        # Find and extract rows from hmagma_genes not in magma_genes
              unique_hmagma <- hmagma_genes %>%
                filter(!(gene_ensg %in% magma_genes$gene_ensg))
        # saving the unique genes to hmagma per cell type      
              save(unique_hmagma, file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4_unique_to_hmagma_wo_og.rda"))
              
       }}

saveWorkbook(wb, file = "~/HMAGMA_Protocol/results/tables/genes/og_magma_genes.xlsx", overwrite = TRUE)

##saving as excel
wb <- createWorkbook()

for(t in 1:length(type)){
  for(c in 1:length(cell)){
load(file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4_unique_to_hmagma_wo_og.rda"))
    
    addWorksheet(wb = wb, sheetName = paste0(type_name[[type[t]]], "_", cell_name[[cell[c]]],"_hmagma"))
    writeData(wb, sheet = paste0(type_name[[type[t]]], "_", cell_name[[cell[c]]],"_hmagma"), unique_hmagma)
    
     }}

saveWorkbook(wb, file = "~/HMAGMA_Protocol/results/tables/genes/unique_to_hmagma_genes_wo_og.xlsx", overwrite = TRUE)

