##MAGMA OUTPUT - FILTERING AND SELECTING RELEVANT GENES
#Loading gene information and reference genome snps
load(file = "/required_files/gene_information.rda") #genes
load(file = "/required_files/snps_g1000.rda") #snps

#Cell names and disease names 
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")

#Directories to save the files 
dir<-"/HMAGMA_output_v3/original/"
outdir<-"/output_annotation_v4/"

#genedir
dir.create(path = "/output_annotation_v4/gene_count_og_magma/")
genedir<-"/output_annotation_v4/gene_count_og_magma/"

#Extracting signigicant genes - AFTER MAGMA 
library(dplyr)
#Looping 
dat<-c()
for(t in 1:length(type)) {

    
    #Loading HMAGMA output for each cell type and disease 
    diseasemat<-read.table(paste0(dir, "og_magma_", type[t], "_gc41.genes.out"), header = T)

    #Calculating FDR (false discovery rate) of the gene p-value
    diseasemat$FDR<-p.adjust(p = diseasemat$P, method = "BH") #adding FDR (false discovery rate) to the genes
    
    #Filtering genes based on the FDR value of 0.01, 0.05, 0.1
    diseasemat_0.01<-diseasemat %>% filter(diseasemat$FDR<=0.01)
    diseasemat_0.05<-diseasemat %>% filter(diseasemat$FDR<=0.05)
    diseasemat_0.1<-diseasemat %>% filter(diseasemat$FDR<=0.1)
    
    # Example ENSG IDs
    ensg_ids <- diseasemat_0.05$GENE
    
    # Get HGNC symbol and description
    gene_info <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol", "description"),
                       filters = "ensembl_gene_id",
                       values = ensg_ids,
                       mart = ensembl)
    
    print(gene_info)
    
    diseasemat_names<-merge(x =diseasemat_0.05, y = genes, by.x="GENE", by.y="gene_id")
    
    #Adding gene names to ensgs for genes with FDR p<0.05
    diseasemat_names<-merge(x =diseasemat_names, y = gene_info, by.x="GENE", by.y="ensembl_gene_id", all.x=TRUE)
    
    #Reformatting and adding column names 
    diseasemat_names<-diseasemat_names[,c(1,19,2,12,13,15,5,8,9,10,18,21)]
    colnames(diseasemat_names)<-c("gene_ensg", "gene_name", "gene_chr", "gene_start", "gene_end",
                                  "gene_strand", "nsnps","gene_zstat", "gene_p", "gene_fdr", "gene_biotype", "gene_description") 
    #Ordering genes by FDR p-value 
    diseasemat_names<-diseasemat_names[order(-diseasemat_names$gene_zstat),]
    
    #Adding a column with cell type and disease type
    try(diseasemat_names$disease<-paste0(type[t]))
    
    #Counting the number of genes generated by HMAGMA filtered by FDR p<0.05
    gene_count<-data.frame(nrow(diseasemat_names))
    rownames(gene_count) <- paste0(type[t], "_count")
    gene_count <- data.frame(names = row.names(gene_count), gene_count) 
    gene_count<- gene_count %>% data.frame(do.call("rbind", strsplit(as.character(gene_count$names), "_", fixed = TRUE)))
    write.table(x = gene_count, file = paste0(genedir, type[t], "_count.txt"), row.names = F, col.names = F, quote = F, sep = "\t")

    save(diseasemat_names, file= paste0(outdir, type[t], "og_magma_fdr005_output_genes_v4.rda"))
    
    
  }

##overlapping magma genes wiht each cell type per disease
library(openxlsx)
outdir<-"/output_annotation_v4/"

cell<- c("Microglia_interactome", "Neuronal_interactome", "Oligo_interactome")
cell_name<-c("Microglia_interactome"="microglia", "Neuronal_interactome" = "neurons", "Oligo_interactome"="oligo")
type<-c("AD_Jansen2019", "AD_Kunkle2019", "PD_Nalls2019proxy", "MS_Andlauer2016", "ALS_Rheenen2021", "SCZ_Trubetskoy2022")
type_name<-c("AD_Jansen2019"="AD_Jansen2019", "AD_Kunkle2019"="AD_Kunkle2019", "PD_Nalls2019proxy"="PD", "MS_Andlauer2016"="MS", "ALS_Rheenen2021"="ALS", "SCZ_Trubetskoy2022"="SCZ")

wb <- createWorkbook()

for(t in 1:length(type)){
  load(file = paste0(paste0(outdir, type[t], "og_magma_fdr005_output_genes_v4.rda")))
       magma_genes<-diseasemat_names
       addWorksheet(wb = wb, sheetName = paste0(type_name[[type[t]]], "_magma"))
       writeData(wb, sheet = paste0(type_name[[type[t]]], "_magma"), magma_genes)
       
       for(c in 1:length(cell)){
         load(file=paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4.rda"))
              hmagma_genes<-diseasemat_names 
              
        # Find and extract rows from hmagma_genes not in magma_genes
              unique_hmagma <- hmagma_genes %>%
                filter(!(gene_ensg %in% magma_genes$gene_ensg))
        # saving the unique genes to hmagma per cell type      
              save(unique_hmagma, file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4_unique_to_hmagma_wo_og.rda"))
              
       }}

saveWorkbook(wb, file = "/results/tables/genes/og_magma_genes.xlsx", overwrite = TRUE)

##saving as excel
wb <- createWorkbook()

for(t in 1:length(type)){
  for(c in 1:length(cell)){
load(file= paste0(outdir, cell[c], "_",  type[t], "_fdr005_output_genes_v4_unique_to_hmagma_wo_og.rda"))
    
    addWorksheet(wb = wb, sheetName = paste0(type_name[[type[t]]], "_", cell_name[[cell[c]]],"_hmagma"))
    writeData(wb, sheet = paste0(type_name[[type[t]]], "_", cell_name[[cell[c]]],"_hmagma"), unique_hmagma)
    
     }}

saveWorkbook(wb, file = "/results/tables/genes/unique_to_hmagma_genes_wo_og.xlsx", overwrite = TRUE)

