#Overlapping promoter and exonic coordinates with g1000 SNPs
```{r}
#BiocManager::install("rtracklayer")
library(rtracklayer)
options(stringsAsFactors = F)
library(GenomicRanges)
library(biomaRt)
library(dplyr)
library(readxl)
library(readr)
library(data.table)
library(plyr)
library(xlsx)
library(tidyr)

```


```{r cars}

setwd("/rds/general/user/aa19618/home/")
gcrh37_41<-rtracklayer::import("~/HMAGMA_Protocol/gencode/gencode.v41lift37.annotation.gtf")
gcrh37_41<-as.data.frame(x=gcrh37_41)
tail(gcrh37_41)

gcrh37_41<-gcrh37_41[1:1000,]

#promoter<-read.table(file = "../HMAGMA/HMAGMA_Protocol/Gencode26_promoter.bed", header = F)
#gene<-read.table(file = "../HMAGMA/HMAGMA_Protocol/Gencode26_gene.bed", header = F)
```

## Generating promoter and exon coordinates

```{r pressure, echo=FALSE}

#extracting required columns (1-seqnames, 2-start, 3-end, 4-width, 5-strand, 6-source, 7-type, 10-gene_id, 11-gene_type, 12-gene_name)
gcrh37_41_edi<-gcrh37_41[,c(1,2,3,4,5,6,7,10,11,12)]

#gene IDs have accession numbers that are removed (https://stackoverflow.com/questions/10617702/remove-part-of-string-after)
gcrh37_41_edi$gene_id<-gsub(pattern = "\\..*", replacement = "", as.character(gcrh37_41_edi$gene_id))

#extracting rows with information for type GENES and saving to be used in HMAGMA annot file making
genes<-gcrh37_41_edi %>% filter(grepl(pattern = 'gene', x = type))
save(genes, file = "~/HMAGMA_Protocol/required_files/gene_information.rda")

#extracting rows with information for type TRANSCRIPT and EXON
transcripts<-gcrh37_41_edi %>% filter(grepl(pattern = 'transcript', x = type))
exons<-gcrh37_41_edi %>% filter(grepl(pattern = 'exon', x = type))

#separating negative and positive strands 
#selecting positive strand
transcripts_positive<- transcripts %>% filter(!strand %in% c("-"))
#selecting negative strand 
transcripts_negative<- transcripts %>% filter(!strand %in% c("+"))

#adding promoter coordinates to NEGATIVE strand 1500 upstream and 500 downstream of the TSS
transcripts_negative$pro_start<-with(data=transcripts_negative, expr = end +1500) %>% as.numeric()
transcripts_negative$pro_end<-with(data=transcripts_negative, expr = end -500) %>% as.numeric()
transcripts_negative<-transcripts_negative[,c(1,2,3,4,5,11,12,6,7,8,9,10)]

#adding promoter coordinates to POSITIVE strand 1500 upstream and 500 downstream of the TSS
transcripts_positive$pro_start<-with(data=transcripts_positive, expr = start -1500) %>% as.numeric()
transcripts_positive$pro_end<-with(data=transcripts_positive, expr = start +500) %>% as.numeric()
transcripts_positive<-transcripts_positive[,c(1,2,3,4,5,11,12,6,7,8,9,10)]

#binding NEGATIVE and POSITIVE strand data 
transcripts_edited<-rbind(transcripts_negative, transcripts_positive) %>% as_data_frame()
#making negative values ZERO (0) since these are probably at the beginning of the DNA sequence 
transcripts_edited[transcripts_edited<0]<-0
#for each promoter make the start<end 
transcripts_edited$start_test<-pmin(transcripts_edited$pro_start, transcripts_edited$pro_end) %>% as.numeric()
transcripts_edited$end_test<-pmax(transcripts_edited$pro_start, transcripts_edited$pro_end) %>% as.numeric()
transcripts_edited_<-transcripts_edited

#selecting columns for promoter file
promoter<-transcripts_edited_[,c("seqnames", "start_test", "end_test", "gene_id")]
#making the start and end positions in the numeric format 
promoter<-transform(promoter, start_test=as.numeric(start_test), end_test=as.numeric(end_test))
sapply(promoter_, mode) #checks the format of each column 

#Naming the columns 
colnames(promoter)<-c("seqnames", "start", "end", "gene_id")

#Selectign the selevant columns in exon and gene dataframes
exon<-exons[,c("seqnames", "start", "end","gene_id")]
gene<-genes[,c("seqnames", "start", "end","gene_id")]

save(promoter,exon,gene, file="required_files/gencode41_37_pro_exo_gene.rda")

```

## Make GenomicRanges objects for exons and promoter

We will use gene definitions from Gencode v41 (grch37): <https://www.gencodegenes.org/human/release_41lift37.html>

```{r generate granges objects from exons and promoters}

load("~/HMAGMA_Protocol/required_files/gencode41_37_pro_exo_gene.rda")

#Making GRanges that will be used in finOverlaps() function 
exonranges <- GRanges(exon[,1],IRanges(exon[,2],exon[,3]),gene=exon[,4])
promoterranges <- GRanges(promoter[,1], IRanges(promoter[,2], promoter[,3]), gene=promoter[,4])

save(exonranges, promoterranges, file="~/HMAGMA_Protocol/required_files/exon_promoranges.rda")
```

## Generate a GenomicRanges object for the SNP annotation

Here, we will use SNP annotation from European ancestry (EUR.bim), which can be downloaded from: <https://ctg.cncr.nl/software/magma>

```{r}

#reading g1000 reference genome from European ancestry and selecting only relevant columns (chr, rsid, position)
g1000<-fread(file = "~/HMAGMA_Protocol/required_files/g1000/g1000_files/g1000_eur.bim", select = c(1,2,4))
snps<-g1000
colnames(snps) <- c("chr","SNP","Position")
#adding chr to the chromosome number 
snps$chr <- sub("^", "chr", snps$chr)

#replacing chr23 with chrX since g1000 reference genome only had chrX data (see in README of g100_eur folder) (THIS STEP WAS SKIPPED)
#snps$chr<-gsub('chr23', 'chrX', snps$chr) 

#making GRances that will be used in findOverlaps() fucntion 
snps <- GRanges(snps$chr, IRanges(snps$Position, snps$Position), rsid=snps$SNP)

save(snps, file="required_files/snps_g1000.rda")

## NOTE: chr23 (referring to chrX) not included when overlapping with findOvelaps() function with exons and introns
```

## Overlap exons and promoters with SNPs
```{r Overlap exons with SNPs} 
load(file = "~/HMAGMA_Protocol/required_files/exon_promoranges.rda")
load(file = "~/HMAGMA_Protocol/required_files/snps_g1000.rda")

#identifying snps located within exons
olap <- findOverlaps(snps,exonranges); #default maxgap=-1L, one range's start/end strictly inside the other, the gap is considered to be -1.
snpexon <- snps[queryHits(olap)];
mcols(snpexon) <- cbind(mcols(snpexon), mcols(exonranges[subjectHits(olap)]))

#identifying snps located within promoters
olap <- findOverlaps(snps,promoterranges);
snpro <- snps[queryHits(olap)];
mcols(snpro) <- cbind(mcols(snpro), mcols(promoterranges[subjectHits(olap)]))

#removing chrX from 
snpexon <- snpexon[seqnames(snpexon)!="chrX"]
snpro <- snpro[seqnames(snpro)!="chrX"]
save(snpro, snpexon, file="required_files/snp_locating_in_exon_promoter_transcript_level_g1000.rda")

load("~/HMAGMA_Protocol/required_files/snp_locating_in_exon_promoter_transcript_level_g1000.rda")
```

## Non-exonic and non-promoter SNPs

```{r}

#removing all the snps that are present within exons
snpranges <- snps[!(snps$rsid %in% snpexon$rsid), ]

#removing all the snps that are present within promoters
snpranges <- snpranges[!(snpranges$rsid %in% snpro$rsid), ]

#saving intronic and intergenic (non exonic, non promoter) snps
save(snpranges,file="required_files/non_exonic_promoter_snp_g1000.rda")
```



